ccl/multiregionccl: TestMultiRegionDataDriven failed

ccl/multiregionccl.TestMultiRegionDataDriven [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_UnitTests_BazelUnitTests/8924823?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_UnitTests_BazelUnitTests/8924823?buildTab=artifacts#/) on master @ [fcea283ebca17a6d923c5d4b0401697438b77dbd](https://github.com/cockroachdb/cockroach/commits/fcea283ebca17a6d923c5d4b0401697438b77dbd):


```
=== RUN   TestMultiRegionDataDriven
I230304 07:16:59.107940 1 (gostd) rand.go:199  [T1] 1  random seed: 4267267524760381632
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/a5ee0b0d8f7b79a0b4e56a682326700f/logTestMultiRegionDataDriven2123913141
    test_log_scope.go:79: use -show-logs to present logs inline
=== CONT  TestMultiRegionDataDriven
    datadriven_test.go:428: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/a5ee0b0d8f7b79a0b4e56a682326700f/logTestMultiRegionDataDriven2123913141
--- FAIL: TestMultiRegionDataDriven (203.33s)
=== RUN   TestMultiRegionDataDriven/secondary_region
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/9326/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/secondary_region:1:
        new-cluster [1 args]
        <no input to command>
        ----
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/9326/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/secondary_region:4:
        exec-sql [1 args]
        SET CLUSTER SETTING server.time_until_store_dead = '25s';
        ----
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/9326/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/secondary_region:8:
        exec-sql [1 args]
        CREATE DATABASE db PRIMARY REGION "us-west-1" REGIONS "us-central-1", "eu-west-1" SECONDARY REGION "us-east-1";
        ----
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/9326/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/secondary_region:12:
        exec-sql [1 args]
        ALTER DATABASE db SURVIVE REGION FAILURE;
        ----
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/9326/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/secondary_region:16:
        exec-sql [1 args]
        CREATE TABLE db.rbt(k INT PRIMARY KEY, v INT) LOCALITY REGIONAL BY TABLE IN "us-west-1";
        ----
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/9326/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/secondary_region:20:
        exec-sql [1 args]
        INSERT INTO db.rbt VALUES (1, 1), (2,2)
        ----
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/9326/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/secondary_region:25:
         
        expected:
        
        found:
        expected leaseholder 2 but got 3
    --- FAIL: TestMultiRegionDataDriven/secondary_region (131.32s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #92235 ccl/multiregionccl: TestMultiRegionDataDriven failed [C-test-failure O-robot T-sql-sessions branch-release-22.2 skipped-test]
</p>
</details>
/cc @cockroachdb/sql-schema
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMultiRegionDataDriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/multiregionccl.TestMultiRegionDataDriven [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8982152?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8982152?buildTab=artifacts#/) on master @ [4dc9e98d8cc1b137ee4cc950a4a42e5895bb890b](https://github.com/cockroachdb/cockroach/commits/4dc9e98d8cc1b137ee4cc950a4a42e5895bb890b):


```
        ALTER DATABASE db DROP REGION "us-west-1"
        ----
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2274/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:187:
        wait-for-zone-config-changes [8 args]
        <no input to command>
        ----
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2274/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:194:
        exec-sql [1 args]
        ALTER DATABASE db SURVIVE REGION FAILURE
        ----
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2274/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:200:
        wait-for-zone-config-changes [5 args]
        <no input to command>
        ----
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2274/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:204:
        sleep-for-follower-read [0 args]
        <no input to command>
        ----
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2274/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:209:
        refresh-range-descriptor-cache [2 args]
        SELECT * FROM db.rbt WHERE k = 1
        ----
        LAG_BY_CLUSTER_SETTING
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2274/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:214:
        refresh-range-descriptor-cache [2 args]
        SELECT * FROM db.rbt WHERE k = 1
        ----
        LAG_BY_CLUSTER_SETTING
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2274/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:219:
        refresh-range-descriptor-cache [2 args]
        SELECT * FROM db.rbt WHERE k = 1
        ----
        LAG_BY_CLUSTER_SETTING
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2274/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:226:
         SELECT * FROM db.rbt AS OF SYSTEM TIME follower_read_timestamp() WHERE k = 1
        expected:
        served locally: true
        served via follower read: true
        
        found:
        recording contains > 1 dist sender send messages
    --- FAIL: TestMultiRegionDataDriven/regional_by_table (42.21s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #92235 ccl/multiregionccl: TestMultiRegionDataDriven failed [C-test-failure O-robot T-sql-sessions branch-release-22.2 skipped-test]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMultiRegionDataDriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/multiregionccl.TestMultiRegionDataDriven [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8998015?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8998015?buildTab=artifacts#/) on master @ [01e14f68f7c757cab1c878cebfecd73c5119046b](https://github.com/cockroachdb/cockroach/commits/01e14f68f7c757cab1c878cebfecd73c5119046b):


```
        ALTER DATABASE db DROP REGION "us-west-1"
        ----
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2955/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:187:
        wait-for-zone-config-changes [8 args]
        <no input to command>
        ----
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2955/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:194:
        exec-sql [1 args]
        ALTER DATABASE db SURVIVE REGION FAILURE
        ----
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2955/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:200:
        wait-for-zone-config-changes [5 args]
        <no input to command>
        ----
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2955/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:204:
        sleep-for-follower-read [0 args]
        <no input to command>
        ----
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2955/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:209:
        refresh-range-descriptor-cache [2 args]
        SELECT * FROM db.rbt WHERE k = 1
        ----
        LAG_BY_CLUSTER_SETTING
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2955/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:214:
        refresh-range-descriptor-cache [2 args]
        SELECT * FROM db.rbt WHERE k = 1
        ----
        LAG_BY_CLUSTER_SETTING
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2955/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:219:
        refresh-range-descriptor-cache [2 args]
        SELECT * FROM db.rbt WHERE k = 1
        ----
        LAG_BY_CLUSTER_SETTING
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2955/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:226:
         SELECT * FROM db.rbt AS OF SYSTEM TIME follower_read_timestamp() WHERE k = 1
        expected:
        served locally: true
        served via follower read: true
        
        found:
        recording contains > 1 dist sender send messages
    --- FAIL: TestMultiRegionDataDriven/regional_by_table (42.16s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #92235 ccl/multiregionccl: TestMultiRegionDataDriven failed [C-test-failure O-robot T-sql-sessions branch-release-22.2 skipped-test]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMultiRegionDataDriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/multiregionccl.TestMultiRegionDataDriven [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9012212?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9012212?buildTab=artifacts#/) on master @ [5b2a5670cbbe895d76602c230390816e783e0caa](https://github.com/cockroachdb/cockroach/commits/5b2a5670cbbe895d76602c230390816e783e0caa):


```
        ALTER DATABASE db DROP REGION "us-west-1"
        ----
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3262/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:187:
        wait-for-zone-config-changes [8 args]
        <no input to command>
        ----
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3262/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:194:
        exec-sql [1 args]
        ALTER DATABASE db SURVIVE REGION FAILURE
        ----
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3262/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:200:
        wait-for-zone-config-changes [5 args]
        <no input to command>
        ----
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3262/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:204:
        sleep-for-follower-read [0 args]
        <no input to command>
        ----
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3262/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:209:
        refresh-range-descriptor-cache [2 args]
        SELECT * FROM db.rbt WHERE k = 1
        ----
        LAG_BY_CLUSTER_SETTING
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3262/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:214:
        refresh-range-descriptor-cache [2 args]
        SELECT * FROM db.rbt WHERE k = 1
        ----
        LAG_BY_CLUSTER_SETTING
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3262/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:219:
        refresh-range-descriptor-cache [2 args]
        SELECT * FROM db.rbt WHERE k = 1
        ----
        LAG_BY_CLUSTER_SETTING
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3262/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:226:
         SELECT * FROM db.rbt AS OF SYSTEM TIME follower_read_timestamp() WHERE k = 1
        expected:
        served locally: true
        served via follower read: true
        
        found:
        recording contains > 1 dist sender send messages
    --- FAIL: TestMultiRegionDataDriven/regional_by_table (43.32s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #92235 ccl/multiregionccl: TestMultiRegionDataDriven failed [C-test-failure O-robot T-sql-sessions branch-release-22.2 skipped-test]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMultiRegionDataDriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/multiregionccl.TestMultiRegionDataDriven [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/9019189?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/9019189?buildTab=artifacts#/) on master @ [95684388bfbb372ac8c7e1a8ebd4a0d6f447e147](https://github.com/cockroachdb/cockroach/commits/95684388bfbb372ac8c7e1a8ebd4a0d6f447e147):


```
        ALTER DATABASE db DROP REGION "us-west-1"
        ----
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/8168/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:187:
        wait-for-zone-config-changes [8 args]
        <no input to command>
        ----
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/8168/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:194:
        exec-sql [1 args]
        ALTER DATABASE db SURVIVE REGION FAILURE
        ----
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/8168/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:200:
        wait-for-zone-config-changes [5 args]
        <no input to command>
        ----
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/8168/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:204:
        sleep-for-follower-read [0 args]
        <no input to command>
        ----
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/8168/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:209:
        refresh-range-descriptor-cache [2 args]
        SELECT * FROM db.rbt WHERE k = 1
        ----
        LAG_BY_CLUSTER_SETTING
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/8168/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:214:
        refresh-range-descriptor-cache [2 args]
        SELECT * FROM db.rbt WHERE k = 1
        ----
        LAG_BY_CLUSTER_SETTING
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/8168/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:219:
        refresh-range-descriptor-cache [2 args]
        SELECT * FROM db.rbt WHERE k = 1
        ----
        LAG_BY_CLUSTER_SETTING
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/8168/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:226:
         SELECT * FROM db.rbt AS OF SYSTEM TIME follower_read_timestamp() WHERE k = 1
        expected:
        served locally: true
        served via follower read: true
        
        found:
        recording contains > 1 dist sender send messages
    --- FAIL: TestMultiRegionDataDriven/regional_by_table (41.59s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #92235 ccl/multiregionccl: TestMultiRegionDataDriven failed [C-test-failure O-robot T-sql-sessions branch-release-22.2 skipped-test]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMultiRegionDataDriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/multiregionccl.TestMultiRegionDataDriven [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9020129?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9020129?buildTab=artifacts#/) on master @ [95684388bfbb372ac8c7e1a8ebd4a0d6f447e147](https://github.com/cockroachdb/cockroach/commits/95684388bfbb372ac8c7e1a8ebd4a0d6f447e147):


```
        ALTER DATABASE db DROP REGION "us-west-1"
        ----
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/30/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:187:
        wait-for-zone-config-changes [8 args]
        <no input to command>
        ----
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/30/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:194:
        exec-sql [1 args]
        ALTER DATABASE db SURVIVE REGION FAILURE
        ----
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/30/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:200:
        wait-for-zone-config-changes [5 args]
        <no input to command>
        ----
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/30/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:204:
        sleep-for-follower-read [0 args]
        <no input to command>
        ----
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/30/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:209:
        refresh-range-descriptor-cache [2 args]
        SELECT * FROM db.rbt WHERE k = 1
        ----
        LAG_BY_CLUSTER_SETTING
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/30/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:214:
        refresh-range-descriptor-cache [2 args]
        SELECT * FROM db.rbt WHERE k = 1
        ----
        LAG_BY_CLUSTER_SETTING
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/30/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:219:
        refresh-range-descriptor-cache [2 args]
        SELECT * FROM db.rbt WHERE k = 1
        ----
        LAG_BY_CLUSTER_SETTING
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/30/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:226:
         SELECT * FROM db.rbt AS OF SYSTEM TIME follower_read_timestamp() WHERE k = 1
        expected:
        served locally: true
        served via follower read: true
        
        found:
        recording contains > 1 dist sender send messages
    --- FAIL: TestMultiRegionDataDriven/regional_by_table (43.02s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #92235 ccl/multiregionccl: TestMultiRegionDataDriven failed [C-test-failure O-robot T-sql-sessions branch-release-22.2 skipped-test]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMultiRegionDataDriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/multiregionccl.TestMultiRegionDataDriven [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9029986?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9029986?buildTab=artifacts#/) on master @ [e4924e2b9be4a36d466beab53a80df9241df4783](https://github.com/cockroachdb/cockroach/commits/e4924e2b9be4a36d466beab53a80df9241df4783):


```
        ALTER DATABASE db DROP REGION "us-west-1"
        ----
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3728/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:187:
        wait-for-zone-config-changes [8 args]
        <no input to command>
        ----
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3728/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:194:
        exec-sql [1 args]
        ALTER DATABASE db SURVIVE REGION FAILURE
        ----
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3728/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:200:
        wait-for-zone-config-changes [5 args]
        <no input to command>
        ----
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3728/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:204:
        sleep-for-follower-read [0 args]
        <no input to command>
        ----
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3728/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:209:
        refresh-range-descriptor-cache [2 args]
        SELECT * FROM db.rbt WHERE k = 1
        ----
        LAG_BY_CLUSTER_SETTING
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3728/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:214:
        refresh-range-descriptor-cache [2 args]
        SELECT * FROM db.rbt WHERE k = 1
        ----
        LAG_BY_CLUSTER_SETTING
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3728/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:219:
        refresh-range-descriptor-cache [2 args]
        SELECT * FROM db.rbt WHERE k = 1
        ----
        LAG_BY_CLUSTER_SETTING
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3728/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:226:
         SELECT * FROM db.rbt AS OF SYSTEM TIME follower_read_timestamp() WHERE k = 1
        expected:
        served locally: true
        served via follower read: true
        
        found:
        recording contains > 1 dist sender send messages
    --- FAIL: TestMultiRegionDataDriven/regional_by_table (42.76s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #92235 ccl/multiregionccl: TestMultiRegionDataDriven failed [C-test-failure O-robot T-sql-sessions branch-release-22.2 skipped-test]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMultiRegionDataDriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/multiregionccl.TestMultiRegionDataDriven [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/9042145?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/9042145?buildTab=artifacts#/) on master @ [36d39aadc5e7b88732e1318363177507193eb52e](https://github.com/cockroachdb/cockroach/commits/36d39aadc5e7b88732e1318363177507193eb52e):


```
        ALTER DATABASE db DROP REGION "us-west-1"
        ----
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3678/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:187:
        wait-for-zone-config-changes [8 args]
        <no input to command>
        ----
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3678/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:194:
        exec-sql [1 args]
        ALTER DATABASE db SURVIVE REGION FAILURE
        ----
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3678/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:200:
        wait-for-zone-config-changes [5 args]
        <no input to command>
        ----
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3678/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:204:
        sleep-for-follower-read [0 args]
        <no input to command>
        ----
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3678/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:209:
        refresh-range-descriptor-cache [2 args]
        SELECT * FROM db.rbt WHERE k = 1
        ----
        LAG_BY_CLUSTER_SETTING
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3678/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:214:
        refresh-range-descriptor-cache [2 args]
        SELECT * FROM db.rbt WHERE k = 1
        ----
        LAG_BY_CLUSTER_SETTING
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3678/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:219:
        refresh-range-descriptor-cache [2 args]
        SELECT * FROM db.rbt WHERE k = 1
        ----
        LAG_BY_CLUSTER_SETTING
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3678/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:226:
         SELECT * FROM db.rbt AS OF SYSTEM TIME follower_read_timestamp() WHERE k = 1
        expected:
        served locally: true
        served via follower read: true
        
        found:
        recording contains > 1 dist sender send messages
    --- FAIL: TestMultiRegionDataDriven/regional_by_table (46.47s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #92235 ccl/multiregionccl: TestMultiRegionDataDriven failed [C-test-failure O-robot T-sql-sessions branch-release-22.2 skipped-test]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMultiRegionDataDriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/multiregionccl.TestMultiRegionDataDriven [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9051124?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9051124?buildTab=artifacts#/) on master @ [4215b86090794a78649aa4463c7bf95d74c57c37](https://github.com/cockroachdb/cockroach/commits/4215b86090794a78649aa4463c7bf95d74c57c37):


```
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3729/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:194:
        exec-sql [1 args]
        ALTER DATABASE db SURVIVE REGION FAILURE
        ----
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3729/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:200:
        wait-for-zone-config-changes [5 args]
        <no input to command>
        ----
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3729/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:204:
        sleep-for-follower-read [0 args]
        <no input to command>
        ----
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3729/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:209:
        refresh-range-descriptor-cache [2 args]
        SELECT * FROM db.rbt WHERE k = 1
        ----
        LAG_BY_CLUSTER_SETTING
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3729/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:214:
        refresh-range-descriptor-cache [2 args]
        SELECT * FROM db.rbt WHERE k = 1
        ----
        LAG_BY_CLUSTER_SETTING
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3729/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:219:
        refresh-range-descriptor-cache [2 args]
        SELECT * FROM db.rbt WHERE k = 1
        ----
        LAG_BY_CLUSTER_SETTING
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3729/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:226:
        trace-sql [1 args]
        SELECT * FROM db.rbt AS OF SYSTEM TIME follower_read_timestamp() WHERE k = 1
        ----
        served locally: true
        served via follower read: true
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3729/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:232:
         SELECT * FROM db.rbt AS OF SYSTEM TIME follower_read_timestamp() WHERE k = 1
        expected:
        served locally: true
        served via follower read: true
        
        found:
        recording contains > 1 dist sender send messages
    --- FAIL: TestMultiRegionDataDriven/regional_by_table (43.29s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #92235 ccl/multiregionccl: TestMultiRegionDataDriven failed [C-test-failure O-robot T-sql-sessions branch-release-22.2 skipped-test]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMultiRegionDataDriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/multiregionccl.TestMultiRegionDataDriven [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/9121581?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/9121581?buildTab=artifacts#/) on master @ [6c99966f604f3521acdb925b9f689529ffd46df3](https://github.com/cockroachdb/cockroach/commits/6c99966f604f3521acdb925b9f689529ffd46df3):


```
        wait-for-zone-config-changes [5 args]
        <no input to command>
        ----
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/11673/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:204:
        sleep-for-follower-read [0 args]
        <no input to command>
        ----
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/11673/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:209:
        refresh-range-descriptor-cache [2 args]
        SELECT * FROM db.rbt WHERE k = 1
        ----
        LAG_BY_CLUSTER_SETTING
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/11673/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:214:
        refresh-range-descriptor-cache [2 args]
        SELECT * FROM db.rbt WHERE k = 1
        ----
        LAG_BY_CLUSTER_SETTING
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/11673/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:219:
        refresh-range-descriptor-cache [2 args]
        SELECT * FROM db.rbt WHERE k = 1
        ----
        LAG_BY_CLUSTER_SETTING
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/11673/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:226:
        trace-sql [1 args]
        SELECT * FROM db.rbt AS OF SYSTEM TIME follower_read_timestamp() WHERE k = 1
        ----
        served locally: true
        served via follower read: true
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/11673/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:232:
        trace-sql [1 args]
        SELECT * FROM db.rbt AS OF SYSTEM TIME follower_read_timestamp() WHERE k = 1
        ----
        served locally: true
        served via follower read: true
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/11673/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:238:
         SELECT * FROM db.rbt AS OF SYSTEM TIME follower_read_timestamp() WHERE k = 1
        expected:
        served locally: true
        served via follower read: true
        
        found:
        recording contains > 1 dist sender send messages
    --- FAIL: TestMultiRegionDataDriven/regional_by_table (77.08s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #92235 ccl/multiregionccl: TestMultiRegionDataDriven failed [C-test-failure O-robot T-sql-sessions branch-release-22.2 skipped-test]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMultiRegionDataDriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/multiregionccl.TestMultiRegionDataDriven [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9131604?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9131604?buildTab=artifacts#/) on master @ [53dbb86acb1d48309530181b94838faf937084d3](https://github.com/cockroachdb/cockroach/commits/53dbb86acb1d48309530181b94838faf937084d3):


```
        ALTER DATABASE db DROP REGION "us-west-1"
        ----
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3359/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:187:
        wait-for-zone-config-changes [8 args]
        <no input to command>
        ----
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3359/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:194:
        exec-sql [1 args]
        ALTER DATABASE db SURVIVE REGION FAILURE
        ----
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3359/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:200:
        wait-for-zone-config-changes [5 args]
        <no input to command>
        ----
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3359/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:204:
        sleep-for-follower-read [0 args]
        <no input to command>
        ----
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3359/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:209:
        refresh-range-descriptor-cache [2 args]
        SELECT * FROM db.rbt WHERE k = 1
        ----
        LAG_BY_CLUSTER_SETTING
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3359/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:214:
        refresh-range-descriptor-cache [2 args]
        SELECT * FROM db.rbt WHERE k = 1
        ----
        LAG_BY_CLUSTER_SETTING
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3359/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:219:
        refresh-range-descriptor-cache [2 args]
        SELECT * FROM db.rbt WHERE k = 1
        ----
        LAG_BY_CLUSTER_SETTING
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3359/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:226:
         SELECT * FROM db.rbt AS OF SYSTEM TIME follower_read_timestamp() WHERE k = 1
        expected:
        served locally: true
        served via follower read: true
        
        found:
        recording contains > 1 dist sender send messages
    --- FAIL: TestMultiRegionDataDriven/regional_by_table (43.93s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #92235 ccl/multiregionccl: TestMultiRegionDataDriven failed [C-test-failure O-robot T-sql-sessions branch-release-22.2 skipped-test]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMultiRegionDataDriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/multiregionccl.TestMultiRegionDataDriven [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9142055?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9142055?buildTab=artifacts#/) on master @ [9acc75317aebfdfe69fb097a8a28e0040c7a67fd](https://github.com/cockroachdb/cockroach/commits/9acc75317aebfdfe69fb097a8a28e0040c7a67fd):


```
=== RUN   TestMultiRegionDataDriven
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/e558fc8050776f4c54ea39ba371b49da/logTestMultiRegionDataDriven1394509355
    test_log_scope.go:79: use -show-logs to present logs inline
=== CONT  TestMultiRegionDataDriven
    datadriven_test.go:428: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/e558fc8050776f4c54ea39ba371b49da/logTestMultiRegionDataDriven1394509355
--- FAIL: TestMultiRegionDataDriven (195.43s)
=== RUN   TestMultiRegionDataDriven/regional_by_table
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/617/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:1:
        new-cluster [1 args]
        <no input to command>
        ----
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/617/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:4:
        exec-sql [1 args]
        CREATE DATABASE db PRIMARY REGION "us-east-1" REGIONS  "us-central-1", "eu-central-1";
        ----
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/617/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:8:
        exec-sql [1 args]
        CREATE TABLE db.rbt(k INT PRIMARY KEY, v INT) LOCALITY REGIONAL BY TABLE IN "us-east-1";
        ----
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/617/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:12:
        exec-sql [1 args]
        INSERT INTO db.rbt VALUES (1, 1), (2,2)
        ----
    datadriven_test.go:129: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/617/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_table:16:
         
        expected:
        
        found:
        expected leaseholder 0 but got 1
    --- FAIL: TestMultiRegionDataDriven/regional_by_table (145.95s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #92235 ccl/multiregionccl: TestMultiRegionDataDriven failed [C-test-failure O-robot T-sql-sessions branch-release-22.2 skipped-test]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMultiRegionDataDriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

I saw a related [failure](https://teamcity.cockroachdb.com/viewLog.html?buildId=9297993&buildTypeId=Cockroach_BazelEssentialCi) on staging (after #99484):
```
=== RUN   TestMultiRegionDataDriven/regional_by_row
    datadriven_test.go:124: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/4186/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_row:1:
        new-cluster [1 args]
        <no input to command>
        ----
    datadriven_test.go:124: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/4186/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_row:4:
        exec-sql [1 args]
        CREATE DATABASE db PRIMARY REGION "us-east-1" REGIONS  "us-west-1", "us-central-1";
        ----
    datadriven_test.go:124: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/4186/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_row:8:
        exec-sql [1 args]
        CREATE TABLE db.rbr(k INT PRIMARY KEY, v INT) LOCALITY REGIONAL BY ROW;
        ----
    datadriven_test.go:124: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/4186/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_row:12:
        exec-sql [1 args]
        INSERT INTO db.rbr (k, v, crdb_region) VALUES (1, 1, 'us-east-1'), (2, 2, 'us-central-1'), (3, 3, 'us-west-1')
        ----
    datadriven_test.go:124: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/4186/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_row:19:
        wait-for-zone-config-changes [8 args]
        <no input to command>
        ----
    datadriven_test.go:124: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/4186/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_row:22:
        wait-for-zone-config-changes [8 args]
        <no input to command>
        ----
    datadriven_test.go:124: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/4186/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_row:25:
         
        expected:
        
        found:
        could not get leaseholder: no valid lease
    --- FAIL: TestMultiRegionDataDriven/regional_by_row (167.01s)
```
It looks different though, right?

ccl/multiregionccl.TestMultiRegionDataDriven [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9681028?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9681028?buildTab=artifacts#/) on master @ [a4ab6b6243f509e62993b4d81855f23470d9901a](https://github.com/cockroachdb/cockroach/commits/a4ab6b6243f509e62993b4d81855f23470d9901a):


```
I230419 06:45:12.572887 51905626 kv/kvserver/replica_rangefeed.go:737  [n4,s4,r10/3:/Table/{6-7}] 2  RangeFeed failed to nudge: retry rangefeed (REASON_NO_LEASEHOLDER)
I230419 06:45:13.169540 685627 kv/kvserver/store_raft.go:656  [T1,n2,s2,r29/5:/Table/2{7-8},raft] 3  raft ready handling: 0.78s [append=0.00s, apply=0.00s, , other=0.78s], wrote []; node might be overloaded
I230419 06:45:13.173784 770649 kv/kvserver/store_raft.go:656  [T1,n3,s3,r54/5:/Table/5{3-4},raft] 4  raft ready handling: 0.57s [append=0.00s, apply=0.00s, , other=0.57s], wrote []; node might be overloaded
I230419 06:45:13.177904 882869 kv/kvserver/store_raft.go:656  [T1,n4,s4,r59/4:/Table/5{8-9},raft] 5  raft ready handling: 1.60s [append=0.00s, apply=0.00s, non-blocking-sync=0.00s, other=1.60s], wrote [append-batch=258 B, append-ent=195 B (1), ]; node might be overloaded
I230419 06:45:13.250178 1117563 kv/kvserver/store_raft.go:656  [T1,n6,s6,r62/2:/{Table/61-Max},raft] 6  raft ready handling: 0.97s [append=0.00s, apply=0.25s, , other=0.73s], wrote [apply=215 B (1)]; node might be overloaded
I230419 06:45:13.182824 685466 kv/kvserver/store_raft.go:656  [T1,n2,s2,r15/6:/Table/1{3-4},raft] 7  raft ready handling: 0.62s [append=0.00s, apply=0.00s, , other=0.62s], wrote []; node might be overloaded
W230419 06:45:13.179753 2848503 2@rpc/clock_offset.go:276  [T1,n2,rnode=4,raddr=127.0.0.1:34617,class=system,rpc] 10  latency jump (prev avg 126.99ms, current 823.65ms)
I230419 06:45:13.183112 880859 kv/kvserver/store_raft.go:656  [T1,n4,s4,r31/3:/{Table/29-NamespaceTab…},raft] 8  raft ready handling: 0.76s [append=0.00s, apply=0.15s, sync=0.00s, other=0.61s], wrote [append-batch=50 B, apply=213 B (1)]; node might be overloaded
I230419 06:45:13.185363 685211 kv/kvserver/store_raft.go:656  [T1,n2,s2,r7/2:/Table/{3-4},raft] 9  raft ready handling: 0.61s [append=0.00s, apply=0.00s, , other=0.61s], wrote []; node might be overloaded
I230419 06:45:13.212894 770654 kv/kvserver/store_raft.go:656  [T1,n3,s3,r60/5:/Table/{59-60},raft] 12  raft ready handling: 0.60s [append=0.00s, apply=0.00s, , other=0.60s], wrote []; node might be overloaded
I230419 06:45:13.197061 685461 kv/kvserver/store_raft.go:656  [T1,n2,s2,r27/6:/Table/2{5-6},raft] 11  raft ready handling: 0.62s [append=0.00s, apply=0.00s, , other=0.62s], wrote []; node might be overloaded
I230419 06:45:13.193617 881260 kv/kvserver/store_raft.go:656  [T1,n4,s4,r44/4:/Table/4{2-3},raft] 13  raft ready handling: 0.58s [append=0.00s, apply=0.00s, , other=0.58s], wrote []; node might be overloaded
I230419 06:45:13.203929 56613 kv/kvserver/store_raft.go:656  [T1,n1,s1,r55/1:/Table/5{4-5},raft] 14  raft ready handling: 0.89s [append=0.00s, apply=0.00s, , other=0.89s], wrote []; node might be overloaded
I230419 06:45:13.236388 685759 kv/kvserver/store_raft.go:656  [T1,n2,s2,r41/5:/Table/{39-40},raft] 15  raft ready handling: 0.66s [append=0.00s, apply=0.00s, , other=0.66s], wrote []; node might be overloaded
I230419 06:45:13.244100 685111 kv/kvserver/store_raft.go:656  [T1,n2,s2,r54/6:/Table/5{3-4},raft] 16  raft ready handling: 0.71s [append=0.00s, apply=0.00s, , other=0.71s], wrote []; node might be overloaded
W230419 06:45:13.239099 2957738 2@rpc/clock_offset.go:276  [T1,n3,rnode=5,raddr=127.0.0.1:44351,class=default,rpc] 18  latency jump (prev avg 141.26ms, current 729.36ms)
I230419 06:45:13.228411 1033983 kv/kvserver/store_raft.go:656  [T1,n5,s5,r61/3:/Table/6{0-1},raft] 17  raft ready handling: 0.81s [append=0.00s, apply=0.00s, sync=0.00s, other=0.81s], wrote [append-batch=316 B, append-ent=215 B (1), ]; node might be overloaded
I230419 06:45:13.247528 882762 kv/kvserver/store_raft.go:656  [T1,n4,s4,r17/3:/Table/1{5-6},raft] 19  raft ready handling: 0.74s [append=0.00s, apply=0.65s, sync=0.00s, other=0.09s], wrote [append-batch=50 B, apply=219 B (1)]; node might be overloaded
I230419 06:45:13.275536 880928 kv/kvserver/store_raft.go:656  [T1,n4,s4,r1/2:/{Min-System/NodeL…},raft] 20  raft ready handling: 0.86s [append=0.00s, apply=0.00s, sync=0.00s, other=0.85s], wrote [append-batch=614 B, append-ent=462 B (2), ]; node might be overloaded
I230419 06:45:13.283981 770675 kv/kvserver/store_raft.go:656  [T1,n3,s3,r48/5:/Table/4{6-7},raft] 22  raft ready handling: 0.65s [append=0.00s, apply=0.00s, , other=0.65s], wrote []; node might be overloaded
I230419 06:45:13.261624 770872 kv/kvserver/store_raft.go:656  [T1,n3,s3,r50/5:/Table/{48-50},raft] 21  raft ready handling: 0.63s [append=0.00s, apply=0.00s, , other=0.63s], wrote []; node might be overloaded
W230419 06:45:13.278913 2973591 2@rpc/clock_offset.go:276  [T1,n5,rnode=4,raddr=127.0.0.1:34617,class=default,rpc] 23  latency jump (prev avg 185.87ms, current 725.16ms)
I230419 06:45:13.291785 56888 kv/kvserver/store_raft.go:656  [T1,n1,s1,r39/1:/Table/3{7-8},raft] 25  raft ready handling: 0.70s [append=0.00s, apply=0.00s, , other=0.70s], wrote []; node might be overloaded
W230419 06:45:13.297360 3052359 2@rpc/clock_offset.go:276  [T1,n6,rnode=3,raddr=127.0.0.1:33145,class=system,rpc] 24  latency jump (prev avg 157.98ms, current 810.64ms)
I230419 06:45:13.302677 685394 kv/kvserver/store_raft.go:656  [T1,n2,s2,r50/6:/Table/{48-50},raft] 26  raft ready handling: 0.73s [append=0.00s, apply=0.00s, non-blocking-sync=0.00s, other=0.73s], wrote [append-batch=278 B, append-ent=215 B (1), ]; node might be overloaded
I230419 06:45:13.303800 685358 kv/kvserver/store_raft.go:656  [T1,n2,s2,r62/6:/{Table/61-Max},raft] 27  raft ready handling: 0.78s [append=0.00s, apply=0.00s, , other=0.78s], wrote []; node might be overloaded
I230419 06:45:13.303265 685430 kv/kvserver/store_raft.go:656  [T1,n2,s2,r3/5:/System/{NodeLive…-tsd},raft] 28  raft ready handling: 0.90s [append=0.00s, apply=0.80s, sync=0.00s, other=0.10s], wrote [append-batch=50 B, apply=201 KiB (2)]; node might be overloaded
I230419 06:45:13.321916 685632 kv/kvserver/store_raft.go:656  [T1,n2,s2,r35/2:/Table/3{3-4},raft] 29  raft ready handling: 0.69s [append=0.00s, apply=0.00s, , other=0.69s], wrote []; node might be overloaded
I230419 06:45:13.328207 770685 kv/kvserver/store_raft.go:656  [T1,n3,s3,r18/5:/Table/1{6-7},raft] 31  raft ready handling: 0.99s [append=0.00s, apply=0.00s, sync=0.00s, other=0.99s], wrote [append-batch=280 B, append-ent=217 B (1), ]; node might be overloaded
I230419 06:45:13.328031 56506 kv/kvserver/store_raft.go:656  [T1,n1,s1,r6/1:/Table/{0-3},raft] 30  raft ready handling: 1.21s [append=0.00s, apply=1.00s, , other=0.21s], wrote [append-batch=50 B, apply=209 B (1)]; node might be overloaded
I230419 06:45:13.328704 685548 kv/kvserver/store_raft.go:656  [T1,n2,s2,r22/2:/Table/2{0-1},raft] 32  raft ready handling: 0.99s [append=0.00s, apply=0.78s, sync=0.00s, other=0.21s], wrote [append-batch=50 B, apply=215 B (1)]; node might be overloaded
I230419 06:45:13.332898 57271 kv/kvserver/replica_rangefeed.go:711  [T1,n1,s1,r10/1:/Table/{6-7}] 33  RangeFeed closed timestamp 1681886686.429943709,0 is behind by 26.902877702s
I230419 06:45:13.330055 1033327 kv/kvserver/store_raft.go:656  [T1,n5,s5,r60/4:/Table/{59-60},raft] 34  raft ready handling: 2.01s [append=0.00s, apply=0.88s, non-blocking-sync=0.00s, other=1.13s], wrote [append-batch=50 B, apply=195 B (1)]; node might be overloaded
I230419 06:45:13.335784 56862 kv/kvserver/store_raft.go:656  [T1,n1,s1,r3/1:/System/{NodeLive…-tsd},raft] 36  raft ready handling: 1.42s [append=0.00s, apply=1.09s, non-blocking-sync=0.00s, other=0.33s], wrote [append-batch=174 B, append-ent=76 B (1), apply=201 KiB (2)]; node might be overloaded
I230419 06:45:13.338350 1034585 kv/kvserver/store_raft.go:656  [T1,n5,s5,r28/4:/Table/2{6-7},raft] 35  raft ready handling: 1.88s [append=0.00s, apply=0.00s, non-blocking-sync=0.00s, other=1.88s], wrote [append-batch=258 B, append-ent=195 B (1), ]; node might be overloaded
I230419 06:45:13.339922 1033717 kv/kvserver/store_raft.go:656  [T1,n5,s5,r43/3:/Table/4{1-2},raft] 38  raft ready handling: 1.97s [append=0.00s, apply=0.87s, non-blocking-sync=0.00s, other=1.10s], wrote [append-batch=12 B, apply=229 B (1)]; node might be overloaded
I230419 06:45:13.338707 57051 kv/kvserver/store_raft.go:656  [T1,n1,s1,r16/1:/Table/1{4-5},raft] 37  raft ready handling: 0.74s [append=0.00s, apply=0.00s, , other=0.74s], wrote []; node might be overloaded
I230419 06:45:13.333039 685106 kv/kvserver/store_raft.go:656  [T1,n2,s2,r36/5:/Table/3{4-5},raft] 39  raft ready handling: 0.91s [append=0.00s, apply=0.82s, non-blocking-sync=0.00s, other=0.09s], wrote [append-batch=50 B, apply=219 B (1)]; node might be overloaded
I230419 06:45:13.343135 880247 kv/kvserver/store_raft.go:656  [T1,n4,s4,r24/3:/Table/2{2-3},raft] 40  raft ready handling: 0.77s [append=0.00s, apply=0.00s, sync=0.00s, other=0.77s], wrote [append-batch=260 B, append-ent=197 B (1), ]; node might be overloaded
W230419 06:45:13.349707 775353 2@rpc/clock_offset.go:276  [T1,n3,rnode=1,raddr=127.0.0.1:33603,class=default,rpc] 42  latency jump (prev avg 176.78ms, current 810.89ms)
I230419 06:45:13.350215 1034549 kv/kvserver/store_raft.go:656  [T1,n5,s5,r10/4:/Table/{6-7},raft] 41  raft ready handling: 1.15s [append=0.00s, apply=0.98s, sync=0.00s, other=0.17s], wrote [append-batch=50 B, apply=209 B (1)]; node might be overloaded
I230419 06:45:13.363698 1034532 kv/kvserver/store_raft.go:656  [T1,n5,s5,r22/3:/Table/2{0-1},raft] 43  raft ready handling: 1.28s [append=0.00s, apply=1.09s, sync=0.00s, other=0.18s], wrote [append-batch=12 B, apply=215 B (1)]; node might be overloaded
I230419 06:45:13.356465 880456 kv/kvserver/store_raft.go:656  [T1,n4,s4,r30/4:/Table/2{8-9},raft] 44  raft ready handling: 2.30s [append=0.00s, apply=0.00s, non-blocking-sync=0.00s, other=2.30s], wrote [append-batch=262 B, append-ent=199 B (1), ]; node might be overloaded
W230419 06:45:13.366096 882987 kv/kvserver/replica_gossip.go:157  [T1,n4,s4,r2/2:/System/NodeLiveness{-Max}] 46  could not acquire lease for range gossip: replica unavailable: (n4,s4):2 unable to serve request to r2:/System/NodeLiveness{-Max} [(n1,s1):1, (n4,s4):2, (n6,s6):3, (n5,s5):4, next=5, gen=6]: closed timestamp: 1681885713.352503723,0 (2023-04-19 06:28:33); raft status: {"id":"2","term":77,"vote":"1","commit":425,"lead":"0","raftState":"StatePreCandidate","applied":425,"progress":{},"leadtransferee":"0"}: operation "probe" timed out after 1m0.196s (given timeout 1m0s): result is ambiguous: after 60.15s of attempting command: context deadline exceeded
I230419 06:45:13.389495 1117462 kv/kvserver/store_raft.go:656  [T1,n6,s6,r14/4:/Table/1{2-3},raft] 45  raft ready handling: 1.93s [append=0.00s, apply=0.00s, non-blocking-sync=0.00s, other=1.93s], wrote [append-batch=50 B, ]; node might be overloaded
I230419 06:45:13.368767 882399 kv/kvserver/store_raft.go:656  [T1,n4,s4,r34/3:/Table/3{2-3},raft] 47  raft ready handling: 0.88s [append=0.00s, apply=0.87s, , other=0.01s], wrote [apply=219 B (1)]; node might be overloaded
--- FAIL: TestMultiRegionDataDriven (1447.84s)
=== RUN   TestMultiRegionDataDriven/global_tables
    testcluster.go:1469: condition failed to evaluate within 45s: rpc error: code = Internal desc = An internal server error has occurred. Please check your CockroachDB logs for more details. Error Code: 40003
    --- FAIL: TestMultiRegionDataDriven/global_tables (1446.73s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #101520 ccl/multiregionccl: TestMultiRegionDataDriven failed [C-test-failure O-robot T-sql-schema branch-release-23.1.0]
- #99563 ccl/multiregionccl: TestMultiRegionDataDriven failed [C-test-failure O-robot T-sql-schema branch-release-23.1]
- #92235 ccl/multiregionccl: TestMultiRegionDataDriven failed [C-test-failure O-robot T-sql-sessions branch-release-22.2 skipped-test]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMultiRegionDataDriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/multiregionccl.TestMultiRegionDataDriven [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9721707?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9721707?buildTab=artifacts#/) on master @ [8124bff03d073f35f4d2b6a2048c7f4417d757d9](https://github.com/cockroachdb/cockroach/commits/8124bff03d073f35f4d2b6a2048c7f4417d757d9):


```
=== RUN   TestMultiRegionDataDriven
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/e558fc8050776f4c54ea39ba371b49da/logTestMultiRegionDataDriven736898380
    test_log_scope.go:79: use -show-logs to present logs inline
=== CONT  TestMultiRegionDataDriven
    datadriven_test.go:428: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/e558fc8050776f4c54ea39ba371b49da/logTestMultiRegionDataDriven736898380
--- FAIL: TestMultiRegionDataDriven (1189.67s)
=== RUN   TestMultiRegionDataDriven/global_tables
    datadriven_test.go:124: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/4552/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/global_tables:1:
        new-cluster [1 args]
        <no input to command>
        ----
    datadriven_test.go:124: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/4552/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/global_tables:4:
         CREATE DATABASE db PRIMARY REGION "us-east-1" REGIONS  "us-west-1", "us-central-1";
        expected:
        
        found:
        pq: rpc error: code = Internal desc = An internal server error has occurred. Please check your CockroachDB logs for more details.
    --- FAIL: TestMultiRegionDataDriven/global_tables (1189.66s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #101520 ccl/multiregionccl: TestMultiRegionDataDriven failed [C-test-failure O-robot T-sql-schema branch-release-23.1.0]
- #99563 ccl/multiregionccl: TestMultiRegionDataDriven failed [C-test-failure O-robot T-sql-schema branch-release-23.1]
- #92235 ccl/multiregionccl: TestMultiRegionDataDriven failed [C-test-failure O-robot T-sql-sessions branch-release-22.2 skipped-test]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMultiRegionDataDriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/multiregionccl.TestMultiRegionDataDriven [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/10045241?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/10045241?buildTab=artifacts#/) on master @ [87c9598a9d59433775764c5a78604c2d3a488ad1](https://github.com/cockroachdb/cockroach/commits/87c9598a9d59433775764c5a78604c2d3a488ad1):


```
=== RUN   TestMultiRegionDataDriven
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/e558fc8050776f4c54ea39ba371b49da/logTestMultiRegionDataDriven867874979
    test_log_scope.go:79: use -show-logs to present logs inline
=== CONT  TestMultiRegionDataDriven
    datadriven_test.go:428: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/e558fc8050776f4c54ea39ba371b49da/logTestMultiRegionDataDriven867874979
--- FAIL: TestMultiRegionDataDriven (163.01s)
=== RUN   TestMultiRegionDataDriven/secondary_region
    datadriven.go:143: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3586/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/secondary_region:1:
        new-cluster [1 args]
        <no input to command>
        ----
    datadriven.go:143: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3586/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/secondary_region:4:
        exec-sql [1 args]
        SET CLUSTER SETTING server.time_until_store_dead = '25s';
        ----
    datadriven.go:143: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3586/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/secondary_region:8:
        exec-sql [1 args]
        CREATE DATABASE db PRIMARY REGION "us-west-1" REGIONS "us-central-1", "eu-west-1" SECONDARY REGION "us-east-1";
        ----
    datadriven.go:143: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3586/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/secondary_region:12:
        exec-sql [1 args]
        ALTER DATABASE db SURVIVE REGION FAILURE;
        ----
    datadriven.go:143: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3586/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/secondary_region:16:
        exec-sql [1 args]
        CREATE TABLE db.rbt(k INT PRIMARY KEY, v INT) LOCALITY REGIONAL BY TABLE IN "us-west-1";
        ----
    datadriven.go:143: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3586/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/secondary_region:20:
        exec-sql [1 args]
        INSERT INTO db.rbt VALUES (1, 1), (2,2)
        ----
    datadriven.go:143: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3586/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/secondary_region:25:
         
        expected:
        
        found:
        expected leaseholder 2 but got 3
    --- FAIL: TestMultiRegionDataDriven/secondary_region (133.08s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #101520 ccl/multiregionccl: TestMultiRegionDataDriven failed [C-test-failure O-robot T-sql-foundations branch-release-23.1.0]
- #99563 ccl/multiregionccl: TestMultiRegionDataDriven failed [C-test-failure O-robot T-sql-foundations branch-release-23.1]
- #92235 ccl/multiregionccl: TestMultiRegionDataDriven failed [C-test-failure O-robot T-sql-foundations branch-release-22.2 skipped-test]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMultiRegionDataDriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/multiregionccl.TestMultiRegionDataDriven [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/10084099?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/10084099?buildTab=artifacts#/) on master @ [a65bca232dfde1862e5aff343e7b1f3a379d6cd6](https://github.com/cockroachdb/cockroach/commits/a65bca232dfde1862e5aff343e7b1f3a379d6cd6):


```
=== RUN   TestMultiRegionDataDriven
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/e558fc8050776f4c54ea39ba371b49da/logTestMultiRegionDataDriven1148872100
    test_log_scope.go:79: use -show-logs to present logs inline
=== CONT  TestMultiRegionDataDriven
    datadriven_test.go:428: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/e558fc8050776f4c54ea39ba371b49da/logTestMultiRegionDataDriven1148872100
--- FAIL: TestMultiRegionDataDriven (158.60s)
=== RUN   TestMultiRegionDataDriven/secondary_region
    datadriven.go:143: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3814/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/secondary_region:1:
        new-cluster [1 args]
        <no input to command>
        ----
    datadriven.go:143: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3814/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/secondary_region:4:
        exec-sql [1 args]
        SET CLUSTER SETTING server.time_until_store_dead = '25s';
        ----
    datadriven.go:143: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3814/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/secondary_region:8:
        exec-sql [1 args]
        CREATE DATABASE db PRIMARY REGION "us-west-1" REGIONS "us-central-1", "eu-west-1" SECONDARY REGION "us-east-1";
        ----
    datadriven.go:143: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3814/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/secondary_region:12:
        exec-sql [1 args]
        ALTER DATABASE db SURVIVE REGION FAILURE;
        ----
    datadriven.go:143: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3814/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/secondary_region:16:
        exec-sql [1 args]
        CREATE TABLE db.rbt(k INT PRIMARY KEY, v INT) LOCALITY REGIONAL BY TABLE IN "us-west-1";
        ----
    datadriven.go:143: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3814/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/secondary_region:20:
        exec-sql [1 args]
        INSERT INTO db.rbt VALUES (1, 1), (2,2)
        ----
    datadriven.go:143: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3814/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/secondary_region:25:
         
        expected:
        
        found:
        expected leaseholder 2 but got 3
    --- FAIL: TestMultiRegionDataDriven/secondary_region (130.71s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #101520 ccl/multiregionccl: TestMultiRegionDataDriven failed [C-test-failure O-robot T-sql-foundations branch-release-23.1.0]
- #99563 ccl/multiregionccl: TestMultiRegionDataDriven failed [C-test-failure O-robot T-sql-foundations branch-release-23.1]
- #92235 ccl/multiregionccl: TestMultiRegionDataDriven failed [C-test-failure O-robot T-sql-foundations branch-release-22.2 skipped-test]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMultiRegionDataDriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/multiregionccl.TestMultiRegionDataDriven [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11109336?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11109336?buildTab=artifacts#/) on master @ [f295bd861a3a427652b19c2254d2401ebb4a3c8e](https://github.com/cockroachdb/cockroach/commits/f295bd861a3a427652b19c2254d2401ebb4a3c8e):

Fatal error:

```
panic: use of Span after Finish. Span: /cockroach.roachpb.Internal/Batch. Finish previously called at: <stack not captured. Set debugUseAfterFinish> [recovered]
	panic: use of Span after Finish. Span: /cockroach.roachpb.Internal/Batch. Finish previously called at: <stack not captured. Set debugUseAfterFinish>

```
Stack: 

```
goroutine 13353321 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).recover(0x3b7e12a?, {0x7985520, 0xc01344e180})
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:230 +0x6a
panic({0x5c4da80, 0xc014fceb80})
	GOROOT/src/runtime/panic.go:884 +0x212
github.com/cockroachdb/cockroach/pkg/util/tracing.(*Span).detectUseAfterFinish(0xb53aab8?)
	github.com/cockroachdb/cockroach/pkg/util/tracing/span.go:182 +0x12e
github.com/cockroachdb/cockroach/pkg/util/tracing.(*Span).RecordingType(0xc05e7f8000)
	github.com/cockroachdb/cockroach/pkg/util/tracing/span.go:442 +0x1e
github.com/cockroachdb/cockroach/pkg/util/tracing.(*Span).IsVerbose(...)
	github.com/cockroachdb/cockroach/pkg/util/tracing/span.go:450
github.com/cockroachdb/cockroach/pkg/util/log.getSpanOrEventLog({0x7985520, 0xc062d56630})
	github.com/cockroachdb/cockroach/pkg/util/log/trace.go:92 +0x3a
github.com/cockroachdb/cockroach/pkg/util/log.vEventf({0x7985520, 0xc062d56630}, 0x0, 0x1, 0x0?, 0x0?, {0x62a3f94, 0x19}, {0x0, 0x0, ...})
	github.com/cockroachdb/cockroach/pkg/util/log/trace.go:212 +0x11f
github.com/cockroachdb/cockroach/pkg/util/log.VEvent(...)
	github.com/cockroachdb/cockroach/pkg/util/log/trace.go:231
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*propBuf).FlushLockedWithRaftGroup(0xc00ea694a0, {0x7985520, 0xc01b8717d0}, {0x79a5298, 0xc043e4e4b0})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_proposal_buf.go:601 +0x12c8
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).handleRaftReadyRaftMuLocked.func2(0x58a0d80?)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_raft.go:775 +0xd4
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).withRaftGroupLocked(0xc00ea68c80, 0x5bc8860?)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_raft.go:1998 +0x4e
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).handleRaftReadyRaftMuLocked(_, {_, _}, {{0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_raft.go:772 +0x317
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).handleRaftReady(_, {_, _}, {{0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_raft.go:728 +0x1b0
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).processReady(0xc006166a80, 0x2?)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_raft.go:689 +0x14b
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*raftSchedulerShard).worker(0xc0046b6690, {0x7985520, 0xc01344e180}, {0x7968c60, 0xc006166a80}, 0xc005607500)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/scheduler.go:416 +0x274
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*raftScheduler).Start.func2({0x7985520?, 0xc01344e180?})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/scheduler.go:321 +0x45
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2()
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:484 +0x146
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:475 +0x43b
```
<details><summary>Log preceding fatal error</summary>
<p>

```
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3768/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_row:8:
        exec-sql [1 args]
        CREATE TABLE db.rbr(k INT PRIMARY KEY, v INT) LOCALITY REGIONAL BY ROW;
        ----
*
* ERROR: a panic has occurred!
* use of Span after Finish. Span: /cockroach.roachpb.Internal/Batch. Finish previously called at: <stack not captured. Set debugUseAfterFinish>
* (1) attached stack trace
*   -- stack trace:
*   | runtime.gopanic
*   | 	GOROOT/src/runtime/panic.go:884
*   | [...repeated from below...]
* Wraps: (2) assertion failure
* Wraps: (3) attached stack trace
*   -- stack trace:
*   | github.com/cockroachdb/cockroach/pkg/util/tracing.(*Span).detectUseAfterFinish
*   | 	github.com/cockroachdb/cockroach/pkg/util/tracing/span.go:182
*   | github.com/cockroachdb/cockroach/pkg/util/tracing.(*Span).RecordingType
*   | 	github.com/cockroachdb/cockroach/pkg/util/tracing/span.go:442
*   | github.com/cockroachdb/cockroach/pkg/util/tracing.(*Span).IsVerbose
*   | 	github.com/cockroachdb/cockroach/pkg/util/tracing/span.go:450
*   | github.com/cockroachdb/cockroach/pkg/util/log.getSpanOrEventLog
*   | 	github.com/cockroachdb/cockroach/pkg/util/log/trace.go:92
*   | github.com/cockroachdb/cockroach/pkg/util/log.vEventf
*   | 	github.com/cockroachdb/cockroach/pkg/util/log/trace.go:212
*   | github.com/cockroachdb/cockroach/pkg/util/log.VEvent
*   | 	github.com/cockroachdb/cockroach/pkg/util/log/trace.go:231
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*propBuf).FlushLockedWithRaftGroup
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_proposal_buf.go:601
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).handleRaftReadyRaftMuLocked.func2
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_raft.go:775
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).withRaftGroupLocked
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_raft.go:1998
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).handleRaftReadyRaftMuLocked
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_raft.go:772
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).handleRaftReady
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_raft.go:728
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).processReady
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_raft.go:689
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*raftSchedulerShard).worker
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/scheduler.go:416
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*raftScheduler).Start.func2
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/scheduler.go:321
*   | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2
*   | 	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:484
*   | runtime.goexit
*   | 	GOROOT/src/runtime/asm_amd64.s:1594
* Wraps: (4) use of Span after Finish. Span: /cockroach.roachpb.Internal/Batch. Finish previously called at: <stack not captured. Set debugUseAfterFinish>
* Error types: (1) *withstack.withStack (2) *assert.withAssertionFailure (3) *withstack.withStack (4) *errutil.leafError
*
```
</p>
</details>
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
, <code>stress=true</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMultiRegionDataDriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/multiregionccl.TestMultiRegionDataDriven [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11115283?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11115283?buildTab=artifacts#/) on master @ [f295bd861a3a427652b19c2254d2401ebb4a3c8e](https://github.com/cockroachdb/cockroach/commits/f295bd861a3a427652b19c2254d2401ebb4a3c8e):

Fatal error:

```
panic: use of Span after Finish. Span: /cockroach.roachpb.Internal/Batch. Finish previously called at: <stack not captured. Set debugUseAfterFinish> [recovered]
	panic: use of Span after Finish. Span: /cockroach.roachpb.Internal/Batch. Finish previously called at: <stack not captured. Set debugUseAfterFinish>

```
Stack: 

```
goroutine 15431521 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).recover(0x3b7e12a?, {0x7985520, 0xc017119dd0})
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:230 +0x6a
panic({0x5c4da80, 0xc02284e220})
	GOROOT/src/runtime/panic.go:884 +0x212
github.com/cockroachdb/cockroach/pkg/util/tracing.(*Span).detectUseAfterFinish(0xb53aab8?)
	github.com/cockroachdb/cockroach/pkg/util/tracing/span.go:182 +0x12e
github.com/cockroachdb/cockroach/pkg/util/tracing.(*Span).RecordingType(0xc03e27e800)
	github.com/cockroachdb/cockroach/pkg/util/tracing/span.go:442 +0x1e
github.com/cockroachdb/cockroach/pkg/util/tracing.(*Span).IsVerbose(...)
	github.com/cockroachdb/cockroach/pkg/util/tracing/span.go:450
github.com/cockroachdb/cockroach/pkg/util/log.getSpanOrEventLog({0x7985520, 0xc00c336030})
	github.com/cockroachdb/cockroach/pkg/util/log/trace.go:92 +0x3a
github.com/cockroachdb/cockroach/pkg/util/log.vEventf({0x7985520, 0xc00c336030}, 0x0, 0x1, 0xd?, 0x0?, {0x62a3f94, 0x19}, {0x0, 0x0, ...})
	github.com/cockroachdb/cockroach/pkg/util/log/trace.go:212 +0x11f
github.com/cockroachdb/cockroach/pkg/util/log.VEvent(...)
	github.com/cockroachdb/cockroach/pkg/util/log/trace.go:231
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*propBuf).FlushLockedWithRaftGroup(0xc00629eda0, {0x7985520, 0xc011c31aa0}, {0x79a5298, 0xc01c0fdea0})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_proposal_buf.go:601 +0x12c8
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).handleRaftReadyRaftMuLocked.func2(0x58a0d80?)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_raft.go:775 +0xd4
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).withRaftGroupLocked(0xc00629e580, 0x5bc8860?)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_raft.go:1998 +0x4e
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).handleRaftReadyRaftMuLocked(_, {_, _}, {{0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_raft.go:772 +0x317
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).handleRaftReady(_, {_, _}, {{0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_raft.go:728 +0x1b0
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).processReady(0xc011521500, 0x2?)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_raft.go:689 +0x14b
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*raftSchedulerShard).worker(0xc016e76d90, {0x7985520, 0xc017119dd0}, {0x7968c60, 0xc011521500}, 0xc0115a8000)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/scheduler.go:416 +0x274
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*raftScheduler).Start.func2({0x7985520?, 0xc017119dd0?})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/scheduler.go:321 +0x45
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2()
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:484 +0x146
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:475 +0x43b
```
<details><summary>Log preceding fatal error</summary>
<p>

```
        ----
        served locally: true
        served via follower read: true
=== RUN   TestMultiRegionDataDriven/regional_by_row
*
* ERROR: a panic has occurred!
* use of Span after Finish. Span: /cockroach.roachpb.Internal/Batch. Finish previously called at: <stack not captured. Set debugUseAfterFinish>
* (1) attached stack trace
*   -- stack trace:
*   | runtime.gopanic
*   | 	GOROOT/src/runtime/panic.go:884
*   | [...repeated from below...]
* Wraps: (2) assertion failure
* Wraps: (3) attached stack trace
*   -- stack trace:
*   | github.com/cockroachdb/cockroach/pkg/util/tracing.(*Span).detectUseAfterFinish
*   | 	github.com/cockroachdb/cockroach/pkg/util/tracing/span.go:182
*   | github.com/cockroachdb/cockroach/pkg/util/tracing.(*Span).RecordingType
*   | 	github.com/cockroachdb/cockroach/pkg/util/tracing/span.go:442
*   | github.com/cockroachdb/cockroach/pkg/util/tracing.(*Span).IsVerbose
*   | 	github.com/cockroachdb/cockroach/pkg/util/tracing/span.go:450
*   | github.com/cockroachdb/cockroach/pkg/util/log.getSpanOrEventLog
*   | 	github.com/cockroachdb/cockroach/pkg/util/log/trace.go:92
*   | github.com/cockroachdb/cockroach/pkg/util/log.vEventf
*   | 	github.com/cockroachdb/cockroach/pkg/util/log/trace.go:212
*   | github.com/cockroachdb/cockroach/pkg/util/log.VEvent
*   | 	github.com/cockroachdb/cockroach/pkg/util/log/trace.go:231
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*propBuf).FlushLockedWithRaftGroup
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_proposal_buf.go:601
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).handleRaftReadyRaftMuLocked.func2
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_raft.go:775
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).withRaftGroupLocked
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_raft.go:1998
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).handleRaftReadyRaftMuLocked
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_raft.go:772
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).handleRaftReady
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_raft.go:728
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).processReady
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_raft.go:689
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*raftSchedulerShard).worker
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/scheduler.go:416
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*raftScheduler).Start.func2
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/scheduler.go:321
*   | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2
*   | 	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:484
*   | runtime.goexit
*   | 	GOROOT/src/runtime/asm_amd64.s:1594
* Wraps: (4) use of Span after Finish. Span: /cockroach.roachpb.Internal/Batch. Finish previously called at: <stack not captured. Set debugUseAfterFinish>
* Error types: (1) *withstack.withStack (2) *assert.withAssertionFailure (3) *withstack.withStack (4) *errutil.leafError
*
```
</p>
</details>
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
, <code>stress=true</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMultiRegionDataDriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/multiregionccl.TestMultiRegionDataDriven [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11121970?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11121970?buildTab=artifacts#/) on master @ [e57e9742146b7fac7f7530fa6224e11121ada92f](https://github.com/cockroachdb/cockroach/commits/e57e9742146b7fac7f7530fa6224e11121ada92f):

Fatal error:

```
panic: use of Span after Finish. Span: /cockroach.roachpb.Internal/Batch. Finish previously called at: <stack not captured. Set debugUseAfterFinish> [recovered]
	panic: use of Span after Finish. Span: /cockroach.roachpb.Internal/Batch. Finish previously called at: <stack not captured. Set debugUseAfterFinish>

```
Stack: 

```
goroutine 15481771 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).recover(0x3b7e9ca?, {0x7985ee0, 0xc019e3ac90})
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:230 +0x6a
panic({0x5c4e240, 0xc0414b3b50})
	GOROOT/src/runtime/panic.go:884 +0x212
github.com/cockroachdb/cockroach/pkg/util/tracing.(*Span).detectUseAfterFinish(0x7985ee0?)
	github.com/cockroachdb/cockroach/pkg/util/tracing/span.go:182 +0x12e
github.com/cockroachdb/cockroach/pkg/util/tracing.(*Span).Tracer(...)
	github.com/cockroachdb/cockroach/pkg/util/tracing/span.go:225
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*replicaDecoder).createTracingSpans(0xc01f5cd0d8, {0x7985ee0, 0xc03ac7c810})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_application_decoder.go:154 +0x4f0
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*replicaDecoder).DecodeAndBind(0xc01f5ccc80?, {0x7985ee0, 0xc03ac7c810}, {0xc043550000?, 0xc?, 0x10?})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_application_decoder.go:63 +0x66
github.com/cockroachdb/cockroach/pkg/kv/kvserver/apply.(*Task).Decode(...)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/apply/task.go:142
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).handleRaftReadyRaftMuLocked(_, {_, _}, {{0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_raft.go:878 +0x73a
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).handleRaftReady(_, {_, _}, {{0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_raft.go:728 +0x1b0
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).processReady(0xc002901500, 0x2?)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_raft.go:689 +0x14b
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*raftSchedulerShard).worker(0xc0141247e0, {0x7985ee0, 0xc019e3ac90}, {0x7969620, 0xc002901500}, 0xc004576a80)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/scheduler.go:416 +0x274
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*raftScheduler).Start.func2({0x7985ee0?, 0xc019e3ac90?})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/scheduler.go:321 +0x45
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2()
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:484 +0x146
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:475 +0x43b
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestMultiRegionDataDriven
    test_log_scope.go:167: test logs captured to: /artifacts/tmp/_tmp/e558fc8050776f4c54ea39ba371b49da/logTestMultiRegionDataDriven2649951294
    test_log_scope.go:81: use -show-logs to present logs inline
=== RUN   TestMultiRegionDataDriven/regional_by_row
*
* ERROR: a panic has occurred!
* use of Span after Finish. Span: /cockroach.roachpb.Internal/Batch. Finish previously called at: <stack not captured. Set debugUseAfterFinish>
* (1) attached stack trace
*   -- stack trace:
*   | runtime.gopanic
*   | 	GOROOT/src/runtime/panic.go:884
*   | [...repeated from below...]
* Wraps: (2) assertion failure
* Wraps: (3) attached stack trace
*   -- stack trace:
*   | github.com/cockroachdb/cockroach/pkg/util/tracing.(*Span).detectUseAfterFinish
*   | 	github.com/cockroachdb/cockroach/pkg/util/tracing/span.go:182
*   | github.com/cockroachdb/cockroach/pkg/util/tracing.(*Span).Tracer
*   | 	github.com/cockroachdb/cockroach/pkg/util/tracing/span.go:225
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*replicaDecoder).createTracingSpans
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_application_decoder.go:154
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*replicaDecoder).DecodeAndBind
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_application_decoder.go:63
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver/apply.(*Task).Decode
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/apply/task.go:142
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).handleRaftReadyRaftMuLocked
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_raft.go:878
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).handleRaftReady
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_raft.go:728
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).processReady
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_raft.go:689
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*raftSchedulerShard).worker
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/scheduler.go:416
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*raftScheduler).Start.func2
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/scheduler.go:321
*   | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2
*   | 	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:484
*   | runtime.goexit
*   | 	GOROOT/src/runtime/asm_amd64.s:1594
* Wraps: (4) use of Span after Finish. Span: /cockroach.roachpb.Internal/Batch. Finish previously called at: <stack not captured. Set debugUseAfterFinish>
* Error types: (1) *withstack.withStack (2) *assert.withAssertionFailure (3) *withstack.withStack (4) *errutil.leafError
*
```
</p>
</details>
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
, <code>stress=true</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMultiRegionDataDriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/multiregionccl.TestMultiRegionDataDriven [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11135246?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11135246?buildTab=artifacts#/) on master @ [fcfa15d9694c0b840ac0c5c1b1489fa90a62efd8](https://github.com/cockroachdb/cockroach/commits/fcfa15d9694c0b840ac0c5c1b1489fa90a62efd8):

Fatal error:

```
panic: use of Span after Finish. Span: /cockroach.roachpb.Internal/Batch. Finish previously called at: <stack not captured. Set debugUseAfterFinish> [recovered]
	panic: use of Span after Finish. Span: /cockroach.roachpb.Internal/Batch. Finish previously called at: <stack not captured. Set debugUseAfterFinish>

```
Stack: 

```
goroutine 16044818 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).recover(0x3b7f56a?, {0x7988bc0, 0xc0209024e0})
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:230 +0x6a
panic({0x5c50960, 0xc02cb605e0})
	GOROOT/src/runtime/panic.go:884 +0x212
github.com/cockroachdb/cockroach/pkg/util/tracing.(*Span).detectUseAfterFinish(0xb540ab8?)
	github.com/cockroachdb/cockroach/pkg/util/tracing/span.go:182 +0x12e
github.com/cockroachdb/cockroach/pkg/util/tracing.(*Span).RecordingType(0xc00fdac400)
	github.com/cockroachdb/cockroach/pkg/util/tracing/span.go:442 +0x1e
github.com/cockroachdb/cockroach/pkg/util/tracing.(*Span).IsVerbose(...)
	github.com/cockroachdb/cockroach/pkg/util/tracing/span.go:450
github.com/cockroachdb/cockroach/pkg/util/log.getSpanOrEventLog({0x7988bc0, 0xc02937dfb0})
	github.com/cockroachdb/cockroach/pkg/util/log/trace.go:92 +0x3a
github.com/cockroachdb/cockroach/pkg/util/log.vEventf({0x7988bc0, 0xc02937dfb0}, 0x0, 0x1, 0xd?, 0x0?, {0x62a71c3, 0x19}, {0x0, 0x0, ...})
	github.com/cockroachdb/cockroach/pkg/util/log/trace.go:212 +0x11f
github.com/cockroachdb/cockroach/pkg/util/log.VEvent(...)
	github.com/cockroachdb/cockroach/pkg/util/log/trace.go:231
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*propBuf).FlushLockedWithRaftGroup(0xc024c414a0, {0x7988bc0, 0xc036256120}, {0x79a8938, 0xc01714dc20})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_proposal_buf.go:601 +0x12c8
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).handleRaftReadyRaftMuLocked.func2(0x58a3ae0?)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_raft.go:775 +0xd4
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).withRaftGroupLocked(0xc024c40c80, 0x5bcb740?)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_raft.go:1998 +0x4e
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).handleRaftReadyRaftMuLocked(_, {_, _}, {{0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_raft.go:772 +0x317
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).handleRaftReady(_, {_, _}, {{0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_raft.go:728 +0x1b0
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).processReady(0xc0220ff500, 0x2?)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_raft.go:689 +0x14b
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*raftSchedulerShard).worker(0xc00d27dea0, {0x7988bc0, 0xc0209024e0}, {0x796c240, 0xc0220ff500}, 0xc00ba09500)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/scheduler.go:416 +0x274
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*raftScheduler).Start.func2({0x7988bc0?, 0xc0209024e0?})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/scheduler.go:321 +0x45
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2()
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:484 +0x146
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:475 +0x43b
```
<details><summary>Log preceding fatal error</summary>
<p>

```
        ----
        served locally: true
        served via follower read: true
=== RUN   TestMultiRegionDataDriven/regional_by_row
*
* ERROR: a panic has occurred!
* use of Span after Finish. Span: /cockroach.roachpb.Internal/Batch. Finish previously called at: <stack not captured. Set debugUseAfterFinish>
* (1) attached stack trace
*   -- stack trace:
*   | runtime.gopanic
*   | 	GOROOT/src/runtime/panic.go:884
*   | [...repeated from below...]
* Wraps: (2) assertion failure
* Wraps: (3) attached stack trace
*   -- stack trace:
*   | github.com/cockroachdb/cockroach/pkg/util/tracing.(*Span).detectUseAfterFinish
*   | 	github.com/cockroachdb/cockroach/pkg/util/tracing/span.go:182
*   | github.com/cockroachdb/cockroach/pkg/util/tracing.(*Span).RecordingType
*   | 	github.com/cockroachdb/cockroach/pkg/util/tracing/span.go:442
*   | github.com/cockroachdb/cockroach/pkg/util/tracing.(*Span).IsVerbose
*   | 	github.com/cockroachdb/cockroach/pkg/util/tracing/span.go:450
*   | github.com/cockroachdb/cockroach/pkg/util/log.getSpanOrEventLog
*   | 	github.com/cockroachdb/cockroach/pkg/util/log/trace.go:92
*   | github.com/cockroachdb/cockroach/pkg/util/log.vEventf
*   | 	github.com/cockroachdb/cockroach/pkg/util/log/trace.go:212
*   | github.com/cockroachdb/cockroach/pkg/util/log.VEvent
*   | 	github.com/cockroachdb/cockroach/pkg/util/log/trace.go:231
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*propBuf).FlushLockedWithRaftGroup
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_proposal_buf.go:601
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).handleRaftReadyRaftMuLocked.func2
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_raft.go:775
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).withRaftGroupLocked
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_raft.go:1998
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).handleRaftReadyRaftMuLocked
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_raft.go:772
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).handleRaftReady
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_raft.go:728
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).processReady
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_raft.go:689
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*raftSchedulerShard).worker
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/scheduler.go:416
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*raftScheduler).Start.func2
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/scheduler.go:321
*   | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2
*   | 	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:484
*   | runtime.goexit
*   | 	GOROOT/src/runtime/asm_amd64.s:1594
* Wraps: (4) use of Span after Finish. Span: /cockroach.roachpb.Internal/Batch. Finish previously called at: <stack not captured. Set debugUseAfterFinish>
* Error types: (1) *withstack.withStack (2) *assert.withAssertionFailure (3) *withstack.withStack (4) *errutil.leafError
*
```
</p>
</details>
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
, <code>stress=true</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMultiRegionDataDriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/multiregionccl.TestMultiRegionDataDriven [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11151381?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11151381?buildTab=artifacts#/) on master @ [8d2d2bf47cb3da1c27baa140a5afb5d28a5ed8f6](https://github.com/cockroachdb/cockroach/commits/8d2d2bf47cb3da1c27baa140a5afb5d28a5ed8f6):

Fatal error:

```
panic: use of Span after Finish. Span: /cockroach.roachpb.Internal/Batch. Finish previously called at: <stack not captured. Set debugUseAfterFinish> [recovered]
	panic: use of Span after Finish. Span: /cockroach.roachpb.Internal/Batch. Finish previously called at: <stack not captured. Set debugUseAfterFinish>

```
Stack: 

```
goroutine 15580019 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).recover(0x3b81c6a?, {0x7993de0, 0xc00dfc5f20})
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:230 +0x6a
panic({0x5c55b80, 0xc06a1892a0})
	GOROOT/src/runtime/panic.go:884 +0x212
github.com/cockroachdb/cockroach/pkg/util/tracing.(*Span).detectUseAfterFinish(0xb54eab8?)
	github.com/cockroachdb/cockroach/pkg/util/tracing/span.go:182 +0x12e
github.com/cockroachdb/cockroach/pkg/util/tracing.(*Span).RecordingType(0xc0083d6400)
	github.com/cockroachdb/cockroach/pkg/util/tracing/span.go:442 +0x1e
github.com/cockroachdb/cockroach/pkg/util/tracing.(*Span).IsVerbose(...)
	github.com/cockroachdb/cockroach/pkg/util/tracing/span.go:450
github.com/cockroachdb/cockroach/pkg/util/log.getSpanOrEventLog({0x7993de0, 0xc009eb5f20})
	github.com/cockroachdb/cockroach/pkg/util/log/trace.go:92 +0x3a
github.com/cockroachdb/cockroach/pkg/util/log.vEventf({0x7993de0, 0xc009eb5f20}, 0x0, 0x1, 0x0?, 0x0?, {0x62ac7d5, 0x19}, {0x0, 0x0, ...})
	github.com/cockroachdb/cockroach/pkg/util/log/trace.go:212 +0x11f
github.com/cockroachdb/cockroach/pkg/util/log.VEvent(...)
	github.com/cockroachdb/cockroach/pkg/util/log/trace.go:231
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*propBuf).FlushLockedWithRaftGroup(0xc019cfc120, {0x7993de0, 0xc01832c5d0}, {0x79b3b58, 0xc019d6b950})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_proposal_buf.go:601 +0x12c8
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).handleRaftReadyRaftMuLocked.func2(0x58a8a40?)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_raft.go:775 +0xd4
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).withRaftGroupLocked(0xc019cfb900, 0x5bd0800?)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_raft.go:1998 +0x4e
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).handleRaftReadyRaftMuLocked(_, {_, _}, {{0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_raft.go:772 +0x317
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).handleRaftReady(_, {_, _}, {{0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_raft.go:728 +0x1b0
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).processReady(0xc00685aa80, 0x2?)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_raft.go:689 +0x14b
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*raftSchedulerShard).worker(0xc02839f180, {0x7993de0, 0xc00dfc5f20}, {0x7977440, 0xc00685aa80}, 0xc003dc5500)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/scheduler.go:416 +0x274
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*raftScheduler).Start.func2({0x7993de0?, 0xc00dfc5f20?})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/scheduler.go:321 +0x45
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2()
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:484 +0x146
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:475 +0x43b
```
<details><summary>Log preceding fatal error</summary>
<p>

```
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1173/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_row:4:
        exec-sql [1 args]
        CREATE DATABASE db PRIMARY REGION "us-east-1" REGIONS  "us-west-1", "us-central-1";
        ----
*
* ERROR: a panic has occurred!
* use of Span after Finish. Span: /cockroach.roachpb.Internal/Batch. Finish previously called at: <stack not captured. Set debugUseAfterFinish>
* (1) attached stack trace
*   -- stack trace:
*   | runtime.gopanic
*   | 	GOROOT/src/runtime/panic.go:884
*   | [...repeated from below...]
* Wraps: (2) assertion failure
* Wraps: (3) attached stack trace
*   -- stack trace:
*   | github.com/cockroachdb/cockroach/pkg/util/tracing.(*Span).detectUseAfterFinish
*   | 	github.com/cockroachdb/cockroach/pkg/util/tracing/span.go:182
*   | github.com/cockroachdb/cockroach/pkg/util/tracing.(*Span).RecordingType
*   | 	github.com/cockroachdb/cockroach/pkg/util/tracing/span.go:442
*   | github.com/cockroachdb/cockroach/pkg/util/tracing.(*Span).IsVerbose
*   | 	github.com/cockroachdb/cockroach/pkg/util/tracing/span.go:450
*   | github.com/cockroachdb/cockroach/pkg/util/log.getSpanOrEventLog
*   | 	github.com/cockroachdb/cockroach/pkg/util/log/trace.go:92
*   | github.com/cockroachdb/cockroach/pkg/util/log.vEventf
*   | 	github.com/cockroachdb/cockroach/pkg/util/log/trace.go:212
*   | github.com/cockroachdb/cockroach/pkg/util/log.VEvent
*   | 	github.com/cockroachdb/cockroach/pkg/util/log/trace.go:231
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*propBuf).FlushLockedWithRaftGroup
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_proposal_buf.go:601
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).handleRaftReadyRaftMuLocked.func2
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_raft.go:775
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).withRaftGroupLocked
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_raft.go:1998
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).handleRaftReadyRaftMuLocked
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_raft.go:772
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).handleRaftReady
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_raft.go:728
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).processReady
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_raft.go:689
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*raftSchedulerShard).worker
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/scheduler.go:416
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*raftScheduler).Start.func2
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/scheduler.go:321
*   | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2
*   | 	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:484
*   | runtime.goexit
*   | 	GOROOT/src/runtime/asm_amd64.s:1594
* Wraps: (4) use of Span after Finish. Span: /cockroach.roachpb.Internal/Batch. Finish previously called at: <stack not captured. Set debugUseAfterFinish>
* Error types: (1) *withstack.withStack (2) *assert.withAssertionFailure (3) *withstack.withStack (4) *errutil.leafError
*
```
</p>
</details>
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
, <code>stress=true</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMultiRegionDataDriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/multiregionccl.TestMultiRegionDataDriven [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11170387?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11170387?buildTab=artifacts#/) on master @ [4fe2a80d81c6fc5a3da3c7c44c5fc38da67e0367](https://github.com/cockroachdb/cockroach/commits/4fe2a80d81c6fc5a3da3c7c44c5fc38da67e0367):

Fatal error:

```
panic: use of Span after Finish. Span: /cockroach.roachpb.Internal/Batch. Finish previously called at: <stack not captured. Set debugUseAfterFinish> [recovered]
	panic: use of Span after Finish. Span: /cockroach.roachpb.Internal/Batch. Finish previously called at: <stack not captured. Set debugUseAfterFinish>

```
Stack: 

```
goroutine 14776304 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).recover(0x3b82f4a?, {0x799ac20, 0xc02059f560})
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:230 +0x6a
panic({0x5c5cbe0, 0xc0209efc40})
	GOROOT/src/runtime/panic.go:884 +0x212
github.com/cockroachdb/cockroach/pkg/util/tracing.(*Span).detectUseAfterFinish(0xb552b80?)
	github.com/cockroachdb/cockroach/pkg/util/tracing/span.go:182 +0x12e
github.com/cockroachdb/cockroach/pkg/util/tracing.(*Span).RecordingType(0xc004db7800)
	github.com/cockroachdb/cockroach/pkg/util/tracing/span.go:442 +0x1e
github.com/cockroachdb/cockroach/pkg/util/tracing.(*Span).IsVerbose(...)
	github.com/cockroachdb/cockroach/pkg/util/tracing/span.go:450
github.com/cockroachdb/cockroach/pkg/util/log.getSpanOrEventLog({0x799ac20, 0xc04fab9440})
	github.com/cockroachdb/cockroach/pkg/util/log/trace.go:92 +0x3a
github.com/cockroachdb/cockroach/pkg/util/log.vEventf({0x799ac20, 0xc04fab9440}, 0x0, 0x1, 0xd?, 0x0?, {0x62b3a9c, 0x19}, {0x0, 0x0, ...})
	github.com/cockroachdb/cockroach/pkg/util/log/trace.go:212 +0x11f
github.com/cockroachdb/cockroach/pkg/util/log.VEvent(...)
	github.com/cockroachdb/cockroach/pkg/util/log/trace.go:231
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*propBuf).FlushLockedWithRaftGroup(0xc02bd78120, {0x799ac20, 0xc0448758c0}, {0x79ba998, 0xc03689eaa0})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_proposal_buf.go:601 +0x12c8
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).handleRaftReadyRaftMuLocked.func2(0x58af620?)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_raft.go:775 +0xd4
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).withRaftGroupLocked(0xc02bd77900, 0x5bd77c0?)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_raft.go:1998 +0x4e
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).handleRaftReadyRaftMuLocked(_, {_, _}, {{0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_raft.go:772 +0x317
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).handleRaftReady(_, {_, _}, {{0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_raft.go:728 +0x1b0
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).processReady(0xc02c2e0a80, 0x2?)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_raft.go:689 +0x14b
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*raftSchedulerShard).worker(0xc0239722a0, {0x799ac20, 0xc02059f560}, {0x797e280, 0xc02c2e0a80}, 0xc02c2e0000)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/scheduler.go:416 +0x274
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*raftScheduler).Start.func2({0x799ac20?, 0xc02059f560?})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/scheduler.go:321 +0x45
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2()
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:484 +0x146
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:475 +0x43b
```
<details><summary>Log preceding fatal error</summary>
<p>

```
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3419/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/multiregionccl/multiregionccl_test_/multiregionccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/multiregionccl/testdata/regional_by_row:4:
        exec-sql [1 args]
        CREATE DATABASE db PRIMARY REGION "us-east-1" REGIONS  "us-west-1", "us-central-1";
        ----
*
* ERROR: a panic has occurred!
* use of Span after Finish. Span: /cockroach.roachpb.Internal/Batch. Finish previously called at: <stack not captured. Set debugUseAfterFinish>
* (1) attached stack trace
*   -- stack trace:
*   | runtime.gopanic
*   | 	GOROOT/src/runtime/panic.go:884
*   | [...repeated from below...]
* Wraps: (2) assertion failure
* Wraps: (3) attached stack trace
*   -- stack trace:
*   | github.com/cockroachdb/cockroach/pkg/util/tracing.(*Span).detectUseAfterFinish
*   | 	github.com/cockroachdb/cockroach/pkg/util/tracing/span.go:182
*   | github.com/cockroachdb/cockroach/pkg/util/tracing.(*Span).RecordingType
*   | 	github.com/cockroachdb/cockroach/pkg/util/tracing/span.go:442
*   | github.com/cockroachdb/cockroach/pkg/util/tracing.(*Span).IsVerbose
*   | 	github.com/cockroachdb/cockroach/pkg/util/tracing/span.go:450
*   | github.com/cockroachdb/cockroach/pkg/util/log.getSpanOrEventLog
*   | 	github.com/cockroachdb/cockroach/pkg/util/log/trace.go:92
*   | github.com/cockroachdb/cockroach/pkg/util/log.vEventf
*   | 	github.com/cockroachdb/cockroach/pkg/util/log/trace.go:212
*   | github.com/cockroachdb/cockroach/pkg/util/log.VEvent
*   | 	github.com/cockroachdb/cockroach/pkg/util/log/trace.go:231
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*propBuf).FlushLockedWithRaftGroup
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_proposal_buf.go:601
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).handleRaftReadyRaftMuLocked.func2
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_raft.go:775
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).withRaftGroupLocked
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_raft.go:1998
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).handleRaftReadyRaftMuLocked
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_raft.go:772
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).handleRaftReady
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_raft.go:728
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).processReady
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_raft.go:689
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*raftSchedulerShard).worker
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/scheduler.go:416
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*raftScheduler).Start.func2
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/scheduler.go:321
*   | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2
*   | 	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:484
*   | runtime.goexit
*   | 	GOROOT/src/runtime/asm_amd64.s:1594
* Wraps: (4) use of Span after Finish. Span: /cockroach.roachpb.Internal/Batch. Finish previously called at: <stack not captured. Set debugUseAfterFinish>
* Error types: (1) *withstack.withStack (2) *assert.withAssertionFailure (3) *withstack.withStack (4) *errutil.leafError
*
```
</p>
</details>
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
, <code>stress=true</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMultiRegionDataDriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/ccl/multiregionccl/testdata/secondary_region
+++ b/pkg/ccl/multiregionccl/testdata/secondary_region
@@ -1,6 +1,3 @@
-skip issue-num=98020
-----
-
 new-cluster localities=us-east-1,us-east-1,us-west-1,us-west-1,us-central-1,us-central-1,us-central-1,eu-west-1,eu-west-1,eu-west-1
 ----
 

--- a/pkg/kv/kvserver/allocator/allocatorimpl/allocator.go
+++ b/pkg/kv/kvserver/allocator/allocatorimpl/allocator.go
@@ -1697,8 +1697,7 @@ func (a Allocator) RebalanceTarget(
 	)
 	var removalConstraintsChecker constraintsCheckFn
 	var rebalanceConstraintsChecker rebalanceConstraintsCheckFn
-	var replicaSetToRebalance, replicasWithExcludedStores []roachpb.ReplicaDescriptor
-	var otherReplicaSet []roachpb.ReplicaDescriptor
+	var replicaSetToRebalance, otherReplicaSet []roachpb.ReplicaDescriptor
 
 	switch t := targetType; t {
 	case VoterTarget:
@@ -1720,7 +1719,6 @@ func (a Allocator) RebalanceTarget(
 		// already have voting replicas as possible candidates. Voting replicas are
 		// supposed to be rebalanced before non-voting replicas, and they do
 		// consider the non-voters' stores as possible candidates.
-		replicasWithExcludedStores = existingVoters
 		otherReplicaSet = existingVoters
 	default:
 		log.KvDistribution.Fatalf(ctx, "unsupported targetReplicaType: %v", t)
@@ -1732,8 +1730,9 @@ func (a Allocator) RebalanceTarget(
 		sl,
 		removalConstraintsChecker,
 		rebalanceConstraintsChecker,
-		replicaSetToRebalance,
-		replicasWithExcludedStores,
+		existingVoters,
+		existingNonVoters,
+		targetType,
 		storePool.GetLocalitiesByStore(replicaSetForDiversityCalc),
 		storePool.IsStoreReadyForRoutineReplicaTransfer,
 		options,
@@ -1779,6 +1778,21 @@ func (a Allocator) RebalanceTarget(
 			log.KvDistribution.VEventf(ctx, 2, "not rebalancing %s to s%d because there are no existing "+
 				"replicas that can be removed", targetType, target.store.StoreID)
 			return zero, zero, "", false
+		}
+
+		// If the target is a necessary non-voter promotion to satisfy some
+		// constraint, then do not attempt to simulate a remove target. We know
+		// that the target can be promoted, whilst another store is demoted (or
+		// removed) in order to satisfy a voter constraint. When every replica is
+		// necessary to satisfy an all-replica, or voter constraint, the simulated
+		// remove replica will not always be the existingCandidate depending on
+		// whether every voter is considered necessary.
+		if target.voterNecessary {
+			removeReplica = roachpb.ReplicationTarget{
+				NodeID:  existingCandidate.store.Node.NodeID,
+				StoreID: existingCandidate.store.StoreID,
+			}
+			break
 		}
 
 		var removeDetails string

--- a/pkg/kv/kvserver/allocator/allocatorimpl/allocator_scorer.go
+++ b/pkg/kv/kvserver/allocator/allocatorimpl/allocator_scorer.go
@@ -693,6 +693,7 @@ type candidate struct {
 	valid           bool
 	fullDisk        bool
 	necessary       bool
+	voterNecessary  bool
 	diversityScore  float64
 	ioOverloaded    bool
 	ioOverloadScore float64
@@ -704,9 +705,11 @@ type candidate struct {
 }
 
 func (c candidate) String() string {
-	str := fmt.Sprintf("s%d, valid:%t, fulldisk:%t, necessary:%t, diversity:%.2f, ioOverloaded: %t, ioOverload: %.2f, converges:%d, "+
-		"balance:%d, hasNonVoter:%t, rangeCount:%d, queriesPerSecond:%.2f",
-		c.store.StoreID, c.valid, c.fullDisk, c.necessary, c.diversityScore, c.ioOverloaded, c.ioOverloadScore, c.convergesScore,
+	str := fmt.Sprintf("s%d, valid:%t, fulldisk:%t, necessary:%t, "+
+		"voterNecessary:%t, diversity:%.2f, ioOverloaded: %t, ioOverload: %.2f, "+
+		"converges:%d, balance:%d, hasNonVoter:%t, rangeCount:%d, queriesPerSecond:%.2f",
+		c.store.StoreID, c.valid, c.fullDisk, c.necessary, c.voterNecessary,
+		c.diversityScore, c.ioOverloaded, c.ioOverloadScore, c.convergesScore,
 		c.balanceScore, c.hasNonVoter, c.rangeCount, c.store.Capacity.QueriesPerSecond)
 	if c.details != "" {
 		return fmt.Sprintf("%s, details:(%s)", str, c.details)
@@ -725,6 +728,9 @@ func (c candidate) compactString() string {
 	}
 	if c.necessary {
 		fmt.Fprintf(&buf, ", necessary:%t", c.necessary)
+	}
+	if c.voterNecessary {
+		fmt.Fprintf(&buf, ", voterNecessary:%t", c.voterNecessary)
 	}
 	if c.diversityScore != 0 {
 		fmt.Fprintf(&buf, ", diversity:%.2f", c.diversityScore)
@@ -771,6 +777,12 @@ func (c candidate) compare(o candidate) float64 {
 			return 400
 		}
 		return -400
+	}
+	if c.voterNecessary != o.voterNecessary {
+		if c.voterNecessary {
+			return 350
+		}
+		return -350
 	}
 	if !scoresAlmostEqual(c.diversityScore, o.diversityScore) {
 		if c.diversityScore > o.diversityScore {
@@ -860,6 +872,7 @@ func (c byScoreAndID) Less(i, j int) bool {
 		c[i].hasNonVoter == c[j].hasNonVoter &&
 		c[i].rangeCount == c[j].rangeCount &&
 		c[i].necessary == c[j].necessary &&
+		c[i].voterNecessary == c[j].voterNecessary &&
 		c[i].fullDisk == c[j].fullDisk &&
 		c[i].ioOverloaded == c[j].ioOverloaded &&
 		c[i].valid == c[j].valid {
@@ -890,6 +903,7 @@ func (cl candidateList) best() candidateList {
 	}
 	for i := 1; i < len(cl); i++ {
 		if cl[i].necessary == cl[0].necessary &&
+			cl[i].voterNecessary == cl[0].voterNecessary &&
 			scoresAlmostEqual(cl[i].diversityScore, cl[0].diversityScore) &&
 			cl[i].convergesScore == cl[0].convergesScore &&
 			cl[i].balanceScore == cl[0].balanceScore &&
@@ -910,6 +924,7 @@ func (cl candidateList) good() candidateList {
 	}
 	for i := 1; i < len(cl); i++ {
 		if cl[i].necessary == cl[0].necessary &&
+			cl[i].voterNecessary == cl[0].voterNecessary &&
 			scoresAlmostEqual(cl[i].diversityScore, cl[0].diversityScore) {
 			continue
 		}
@@ -955,6 +970,7 @@ func (cl candidateList) worst() candidateList {
 	// Find the worst constraint/locality/converges/balanceScore values.
 	for i := len(cl) - 2; i >= 0; i-- {
 		if cl[i].necessary == cl[len(cl)-1].necessary &&
+			cl[i].voterNecessary == cl[len(cl)-1].voterNecessary &&
 			scoresAlmostEqual(cl[i].diversityScore, cl[len(cl)-1].diversityScore) &&
 			cl[i].convergesScore == cl[len(cl)-1].convergesScore &&
 			cl[i].balanceScore == cl[len(cl)-1].balanceScore {
@@ -1476,7 +1492,8 @@ func rankedCandidateListForRebalancing(
 	allStores storepool.StoreList,
 	removalConstraintsChecker constraintsCheckFn,
 	rebalanceConstraintsChecker rebalanceConstraintsCheckFn,
-	existingReplicasForType, replicasOnExemptedStores []roachpb.ReplicaDescriptor,
+	existingVotingReplicas, existingNonVotingReplicas []roachpb.ReplicaDescriptor,
+	targetType TargetReplicaType,
 	existingStoreLocalities map[roachpb.StoreID]roachpb.Locality,
 	isStoreValidForRoutineReplicaTransfer func(context.Context, roachpb.StoreID) bool,
 	options ScorerOptions,
@@ -1484,6 +1501,13 @@ func rankedCandidateListForRebalancing(
 ) []rebalanceOptions {
 	// 1. Determine whether existing replicas are valid and/or necessary.
 	existingStores := make(map[roachpb.StoreID]candidate)
+	var existingReplicasForType []roachpb.ReplicaDescriptor
+	if targetType == VoterTarget {
+		existingReplicasForType = existingVotingReplicas
+	} else {
+		existingReplicasForType = existingNonVotingReplicas
+	}
+
 	var needRebalanceFrom bool
 	curDiversityScore := RangeDiversityScore(existingStoreLocalities)
 	for _, store := range allStores.Stores {
@@ -1563,7 +1587,6 @@ func rankedCandidateListForRebalancing(
 			if store.StoreID == existing.store.StoreID {
 				continue
 			}
-
 			// Ignore any stores on dead nodes or stores that contain any of the
 			// replicas within `replicasOnExemptedStores`.
 			if !isStoreValidForRoutineReplicaTransfer(ctx, store.StoreID) {
@@ -1577,24 +1600,43 @@ func rankedCandidateListForRebalancing(
 				continue
 			}
 
-			// `replsOnExemptedStores` is used during non-voting replica rebalancing
-			// to ignore stores that already have a voting replica for the same range.
-			// When rebalancing a voter, `replsOnExemptedStores` is empty since voters
-			// are allowed to "displace" non-voting replicas (we correctly turn such
-			// actions into non-voter promotions, see replicationChangesForRebalance()).
-			var exempted bool
-			for _, replOnExemptedStore := range replicasOnExemptedStores {
-				if store.StoreID == replOnExemptedStore.StoreID {
-					log.KvDistribution.VEventf(
-						ctx,
-						6,
-						"s%d is not a possible rebalance candidate for non-voters because it already has a voter of the range; ignoring",
-						store.StoreID,
-					)
-					exempted = true
-					break
+			var exempted, promotionCandidate bool
+			if targetType == NonVoterTarget {
+				// During non-voting replica rebalancing we ignore stores that already
+				// have a voting replica for the same range. Voters are allowed to
+				// "displace" non-voting replicas (we correctly turn such actions into
+				// non-voter promotions, see replicationChangesForRebalance()).
+				for _, replOnExemptedStore := range existingVotingReplicas {
+					if store.StoreID == replOnExemptedStore.StoreID {
+						log.KvDistribution.VEventf(
+							ctx,
+							6,
+							"s%d is not a possible rebalance candidate for non-voters because it already has a voter of the range; ignoring",
+							store.StoreID,
+						)
+						exempted = true
+						break
+					}
 				}
+			} else if targetType == VoterTarget {
+				// During voting replica rebalancing, we check whether the candidate
+				// already has a non-voting replica, if so it may be necessary to
+				// promote this candidate to a voter, in order to satisfy a voter
+				// constraint (assume there are already a correct number of voting and
+				// non-voting replicas).
+				for _, repl := range existingNonVotingReplicas {
+					if store.StoreID == repl.StoreID {
+						if repl.Type == roachpb.NON_VOTER {
+							promotionCandidate = true
+							break
+						}
+					}
+				}
+			} else {
+				log.KvDistribution.Fatalf(ctx,
+					"unsupported targetReplicaType: %v", targetType)
 			}
+
 			if exempted {
 				continue
 			}
@@ -1604,20 +1646,20 @@ func rankedCandidateListForRebalancing(
 			// this stage, in additon to hard checks and validation.
 			// TODO(kvoli,ayushshah15): Refactor this to make it harder to
 			// inadvertently break the invariant above,
-			constraintsOK, necessary := rebalanceConstraintsChecker(store, existing.store)
+			constraintsOK, necessary, voterNecessary := rebalanceConstraintsChecker(store, existing.store)
 			diversityScore := diversityRebalanceFromScore(
 				store, existing.store.StoreID, existingStoreLocalities)
 			cand := candidate{
 				store:          store,
 				valid:          constraintsOK,
 				necessary:      necessary,
+				voterNecessary: promotionCandidate && voterNecessary,
 				fullDisk:       !options.getDiskOptions().maxCapacityCheck(store),
 				diversityScore: diversityScore,
 			}
 			if !cand.less(existing) {
 				// If `cand` is not worse than `existing`, add it to the list.
 				comparableCands = append(comparableCands, cand)
-
 				if !needRebalanceFrom && !needRebalanceTo && existing.less(cand) {
 					needRebalanceTo = true
 					log.KvDistribution.VEventf(ctx, 2,
@@ -1844,7 +1886,9 @@ type constraintsCheckFn func(roachpb.StoreDescriptor) (valid, necessary bool)
 // rebalanceConstraintsCheckFn determines whether `toStore` is a valid and/or
 // necessary replacement candidate for `fromStore` (which must contain an
 // existing replica).
-type rebalanceConstraintsCheckFn func(toStore, fromStore roachpb.StoreDescriptor) (valid, necessary bool)
+type rebalanceConstraintsCheckFn func(toStore, fromStore roachpb.StoreDescriptor) (
+	valid, necessary, voterNecessary bool,
+)
 
 // voterConstraintsCheckerForAllocation returns a constraintsCheckFn that
 // determines whether a candidate for a new voting replica is valid and/or
@@ -1919,11 +1963,16 @@ func nonVoterConstraintsCheckerForRemoval(
 func voterConstraintsCheckerForRebalance(
 	overallConstraints, voterConstraints constraint.AnalyzedConstraints,
 ) rebalanceConstraintsCheckFn {
-	return func(toStore, fromStore roachpb.StoreDescriptor) (valid, necessary bool) {
+	return func(toStore, fromStore roachpb.StoreDescriptor) (valid, necessary, voterNecessary bool) {
 		overallConstraintsOK, necessaryOverall := rebalanceFromConstraintsCheck(toStore, fromStore, overallConstraints)
 		voterConstraintsOK, necessaryForVoters := rebalanceFromConstraintsCheck(toStore, fromStore, voterConstraints)
 
-		return overallConstraintsOK && voterConstraintsOK, necessaryOverall || necessaryForVoters
+		// If toStore is necessary to satisfy a voter constraint, whilst fromStore
+		// is not necessary to satisfy a voter constraint, then also include that
+		// this is a voterNecessary rebalance.
+		_, removeNecessaryForVoters := removeConstraintsCheck(fromStore, voterConstraints)
+		return overallConstraintsOK && voterConstraintsOK, necessaryOverall || necessaryForVoters,
+			necessaryForVoters && !removeNecessaryForVoters
 	}
 }
 
@@ -1933,8 +1982,9 @@ func voterConstraintsCheckerForRebalance(
 func nonVoterConstraintsCheckerForRebalance(
 	overallConstraints constraint.AnalyzedConstraints,
 ) rebalanceConstraintsCheckFn {
-	return func(toStore, fromStore roachpb.StoreDescriptor) (valid, necessary bool) {
-		return rebalanceFromConstraintsCheck(toStore, fromStore, overallConstraints)
+	return func(toStore, fromStore roachpb.StoreDescriptor) (valid, necessary, necessaryPromo bool) {
+		valid, necessary = rebalanceFromConstraintsCheck(toStore, fromStore, overallConstraints)
+		return valid, necessary, false /* voterNecessary */
 	}
 }
 

--- a/pkg/kv/kvserver/allocator/allocatorimpl/allocator_scorer_test.go
+++ b/pkg/kv/kvserver/allocator/allocatorimpl/allocator_scorer_test.go
@@ -1248,6 +1248,7 @@ func TestShouldRebalanceDiversity(t *testing.T) {
 			rebalanceConstraintsChecker,
 			replicas,
 			nil,
+			VoterTarget,
 			existingStoreLocalities,
 			func(context.Context, roachpb.StoreID) bool { return true },
 			options,

--- a/pkg/kv/kvserver/allocator/allocatorimpl/allocator_test.go
+++ b/pkg/kv/kvserver/allocator/allocatorimpl/allocator_test.go
@@ -3759,6 +3759,7 @@ func TestAllocateCandidatesExcludeNonReadyNodes(t *testing.T) {
 				rebalanceConstraintsChecker,
 				existingRepls,
 				nil,
+				VoterTarget,
 				sp.GetLocalitiesByStore(existingRepls),
 				sp.IsStoreReadyForRoutineReplicaTransfer,
 				a.ScorerOptions(ctx),
@@ -5524,6 +5525,7 @@ func TestRebalanceCandidatesNumReplicasConstraints(t *testing.T) {
 			rebalanceConstraintsChecker,
 			existingRepls,
 			nil,
+			VoterTarget,
 			sp.GetLocalitiesByStore(existingRepls),
 			func(context.Context, roachpb.StoreID) bool { return true },
 			a.ScorerOptions(ctx),
@@ -9146,4 +9148,147 @@ func TestingQPSLoadScorerOptions(
 		RebalanceImpact:              MakeQPSOnlyDim(qpsPerReplica),
 	}
 	return options
+}
+
+// TestAllocatorRebalanceTargetVoterConstraintUnsatisfied exercises the
+// promotion path in allocator.RebalanceVoter. It was possible previously for a
+// satisfiable voter constraint to remain unsatisfied indefinitely (#106559).
+func TestAllocatorRebalanceTargetVoterConstraintUnsatisfied(t *testing.T) {
+	ctx := context.Background()
+	storesFn := func(storeRegions ...string) []*roachpb.StoreDescriptor {
+		var descs []*roachpb.StoreDescriptor
+		for i, storeRegion := range storeRegions {
+			descs = append(descs, &roachpb.StoreDescriptor{
+				StoreID: roachpb.StoreID(i + 1),
+				Node: roachpb.NodeDescriptor{
+					NodeID: roachpb.NodeID(i + 1),
+					Locality: roachpb.Locality{Tiers: []roachpb.Tier{
+						{Key: "region", Value: storeRegion},
+					}},
+				},
+				// Leave the stats as mostly the same to avoid complicating examples.
+				Capacity: roachpb.StoreCapacity{RangeCount: 0, Capacity: 200, Available: 200},
+			})
+
+		}
+		return descs
+	}
+
+	regionConjunctionFn := func(regionName string, numReplicas int32) roachpb.ConstraintsConjunction {
+		return roachpb.ConstraintsConjunction{
+			NumReplicas: numReplicas,
+			Constraints: []roachpb.Constraint{
+				{
+					Type:  roachpb.Constraint_REQUIRED,
+					Key:   "region",
+					Value: regionName,
+				},
+			},
+		}
+	}
+
+	stopper, g, sp, a, _ := CreateTestAllocator(ctx, 6, false /* deterministic */)
+	defer stopper.Stop(ctx)
+	storeLocalities := []string{"a", "a", "b", "b", "c", "c"}
+	gossiputil.NewStoreGossiper(g).GossipStores(storesFn(storeLocalities...), t)
+
+	testCases := []struct {
+		name                            string
+		existingVoters                  []roachpb.ReplicaDescriptor
+		existingNonVoters               []roachpb.ReplicaDescriptor
+		spanConfig                      roachpb.SpanConfig
+		expectedPromote, expectedDemote roachpb.ReplicationTarget
+		expectNoop                      bool
+	}{
+		{
+			// Unsatisfied voter constraint (a:2) requires the non-voter in locality
+			// a to be promoted.
+			name:              "[a(non-voter),a,b,b] -> [a,a,b,b(non-voter)]",
+			existingVoters:    replicas(2, 3, 4),
+			existingNonVoters: replicas(1),
+			spanConfig: roachpb.SpanConfig{
+				NumReplicas: 4,
+				NumVoters:   3,
+				Constraints: []roachpb.ConstraintsConjunction{
+					regionConjunctionFn("a", 2), regionConjunctionFn("b", 2),
+				},
+				VoterConstraints: []roachpb.ConstraintsConjunction{
+					regionConjunctionFn("a", 2), regionConjunctionFn("b", 1),
+				},
+			},
+			expectedPromote: roachpb.ReplicationTarget{NodeID: 1, StoreID: 1},
+			expectedDemote:  roachpb.ReplicationTarget{NodeID: 4, StoreID: 4},
+		},
+		{
+			// Similar to above, except the additional voter region (c) has higher
+			// diversity than the above example. This shouldn't affect the promotion,
+			// which still takes place as higher priority than diversity.
+			name:              "[a(non-voter),a,b,b,c] -> [a,a,b,b,c(non-voter)]",
+			existingVoters:    replicas(2, 3, 4, 5),
+			existingNonVoters: replicas(1),
+			spanConfig: roachpb.SpanConfig{
+				NumReplicas: 5,
+				NumVoters:   4,
+				Constraints: []roachpb.ConstraintsConjunction{
+					regionConjunctionFn("a", 2), regionConjunctionFn("b", 2), regionConjunctionFn("c", 1),
+				},
+				VoterConstraints: []roachpb.ConstraintsConjunction{
+					regionConjunctionFn("a", 2), regionConjunctionFn("b", 2),
+				},
+			},
+			expectedPromote: roachpb.ReplicationTarget{NodeID: 1, StoreID: 1},
+			expectedDemote:  roachpb.ReplicationTarget{NodeID: 5, StoreID: 5},
+		},
+		{
+			// The constraints and voter constraints are satisfied, there's no action
+			// to take here.
+			name:              "[a,a,b,b(non-voter)] -> no-op",
+			existingVoters:    replicas(1, 2, 3),
+			existingNonVoters: replicas(4),
+			spanConfig: roachpb.SpanConfig{
+				NumReplicas: 4,
+				NumVoters:   3,
+				Constraints: []roachpb.ConstraintsConjunction{
+					regionConjunctionFn("a", 2), regionConjunctionFn("b", 2),
+				},
+				VoterConstraints: []roachpb.ConstraintsConjunction{
+					regionConjunctionFn("a", 2), regionConjunctionFn("b", 1),
+				},
+			},
+			expectNoop: true,
+		},
+
+		{
+			// There are no constraints to satisfy, should be a no-op.
+			name:              "[a,a(non-voter),b,b(non-voter),c] -> no-op",
+			existingVoters:    replicas(1, 3, 5),
+			existingNonVoters: replicas(2, 4),
+			spanConfig: roachpb.SpanConfig{
+				NumReplicas: 5,
+				NumVoters:   3,
+			},
+			expectNoop: true,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			for i := range tc.existingNonVoters {
+				tc.existingNonVoters[i].Type = roachpb.NON_VOTER
+			}
+			promote, demote, details, ok := a.RebalanceVoter(
+				ctx, sp, &tc.spanConfig, nil, tc.existingVoters, tc.existingNonVoters,
+				allocator.RangeUsageInfo{}, storepool.StoreFilterNone, a.ScorerOptions(ctx))
+			if tc.expectNoop {
+				require.False(t, ok, "expected ok to be false, details=%v", details)
+				require.Equal(t, roachpb.ReplicationTarget{}, promote)
+				require.Equal(t, roachpb.ReplicationTarget{}, demote)
+			} else {
+				require.Equal(t, tc.expectedPromote, promote, "locality promoted=%s, expected=%s, details=%s",
+					storeLocalities[promote.StoreID-1], storeLocalities[tc.expectedPromote.StoreID-1], details)
+				require.Equal(t, tc.expectedDemote, demote, "locality demoted=%s, expected=%s, details=%s",
+					storeLocalities[demote.StoreID-1], storeLocalities[tc.expectedDemote.StoreID-1], details)
+			}
+		})
+	}
 }

--- a/pkg/kv/kvserver/asim/event/mutation_event.go
+++ b/pkg/kv/kvserver/asim/event/mutation_event.go
@@ -49,10 +49,18 @@ type SetCapacityOverrideEvent struct {
 	CapacityOverride state.CapacityOverride
 }
 
+// SetNodeLocalityEvent represents a mutation event responsible for updating
+// the locality of a node identified by NodeID.
+type SetNodeLocalityEvent struct {
+	NodeID         state.NodeID
+	LocalityString string
+}
+
 var _ Event = &SetSpanConfigEvent{}
 var _ Event = &AddNodeEvent{}
 var _ Event = &SetNodeLivenessEvent{}
 var _ Event = &SetCapacityOverrideEvent{}
+var _ Event = &SetNodeLocalityEvent{}
 
 func (se SetSpanConfigEvent) Func() EventFunc {
 	return MutationFunc(func(ctx context.Context, s state.State) {
@@ -108,4 +116,21 @@ func (sce SetCapacityOverrideEvent) Func() EventFunc {
 
 func (sce SetCapacityOverrideEvent) String() string {
 	return fmt.Sprintf("set capacity override event with storeID=%d, capacity_override=%v", sce.StoreID, sce.CapacityOverride)
+}
+
+func (sne SetNodeLocalityEvent) Func() EventFunc {
+	return MutationFunc(func(ctx context.Context, s state.State) {
+		log.Infof(ctx, "setting node locality %v", sne.LocalityString)
+		if sne.LocalityString != "" {
+			var locality roachpb.Locality
+			if err := locality.Set(sne.LocalityString); err != nil {
+				panic(fmt.Sprintf("unable to set node locality %s", err.Error()))
+			}
+			s.SetNodeLocality(sne.NodeID, locality)
+		}
+	})
+}
+
+func (sne SetNodeLocalityEvent) String() string {
+	return fmt.Sprintf("set node locality event with nodeID=%d, locality=%v", sne.NodeID, sne.LocalityString)
 }

--- a/pkg/kv/kvserver/asim/tests/datadriven_simulation_test.go
+++ b/pkg/kv/kvserver/asim/tests/datadriven_simulation_test.go
@@ -47,7 +47,7 @@ import (
 //     simulation. The default values are: rw_ratio=0 rate=0 min_block=1
 //     max_block=1 min_key=1 max_key=10_000 access_skew=false.
 //
-//   - "ken_cluster" [nodes=<int>] [stores_per_node=<int>]
+//   - "gen_cluster" [nodes=<int>] [stores_per_node=<int>]
 //     Initialize the cluster generator parameters. On the next call to eval,
 //     the cluster generator is called to create the initial state used in the
 //     simulation. The default values are: nodes=3 stores_per_node=1.
@@ -73,6 +73,11 @@ import (
 //     Set the liveness status of the node with ID NodeID. This applies at the
 //     start of the simulation or with some delay after the simulation starts,
 //     if specified.
+//
+//   - set_locality node=<int> [delay=<duration] locality=string
+//     Sets the locality of the node with ID NodeID. This applies at the start
+//     of the simulation or with some delay after the simulation stats, if
+//     specified.
 //
 //   - add_node: [stores=<int>] [locality=<string>] [delay=<duration>]
 //     Add a node to the cluster after initial generation with some delay,
@@ -283,6 +288,19 @@ func TestDataDriven(t *testing.T) {
 				eventGen.ScheduleEvent(settingsGen.Settings.StartTime, delay, event.SetNodeLivenessEvent{
 					NodeId:         state.NodeID(nodeID),
 					LivenessStatus: livenessStatus,
+				})
+				return ""
+			case "set_locality":
+				var nodeID int
+				var localityString string
+				var delay time.Duration
+				scanArg(t, d, "node", &nodeID)
+				scanArg(t, d, "locality", &localityString)
+				scanIfExists(t, d, "delay", &delay)
+
+				eventGen.ScheduleEvent(settingsGen.Settings.StartTime, delay, event.SetNodeLocalityEvent{
+					NodeID:         state.NodeID(nodeID),
+					LocalityString: localityString,
 				})
 				return ""
 			case "set_capacity":

--- a/pkg/kv/kvserver/asim/tests/testdata/non_rand/example_conformance
+++ b/pkg/kv/kvserver/asim/tests/testdata/non_rand/example_conformance
@@ -1,0 +1,64 @@
+# This test reproduces #106559 in a simpler format, where a (satisfiable) voter
+# constraint will never be satisfied if every existing replica is necessary to
+# satisfy either a voter or non-voter constraint. See the topology below.
+gen_cluster nodes=5
+----
+
+# TODO(kvoli): Update gen_cluster to take in localities for this purpose.
+set_locality node=1 locality=region=a
+----
+
+set_locality node=2 locality=region=a
+----
+
+set_locality node=3 locality=region=b
+----
+
+set_locality node=4 locality=region=b
+----
+
+set_locality node=5 locality=region=b
+----
+
+# Generate 10 ranges, one replica will be placed on each node initially.
+gen_ranges ranges=5 repl_factor=4
+----
+
+# Update the span config to require a non-voter in the `region=a` locality.
+# There should be the following replica localities for voters and non-voters:
+#   voters      [a,b,b]
+#   non_voters  [a]
+set_span_config 
+[0,10000): num_replicas=4 num_voters=3 constraints={'+region=a':2,'+region=b':2} voter_constraints={'+region=a':1,'+region=b':2}
+----
+
+# Switch the span config to require the non-voter be placed in the 'region=b'
+# locality. This will require a swap between one of the voting replicas in b,
+# and the non-voter a.
+set_span_config delay=5m
+[0,10000): num_replicas=4 num_voters=3 constraints={'+region=a':2,'+region=b':2} voter_constraints={'+region=a':2,'+region=b':1}
+----
+
+assertion type=conformance under=0 over=0 unavailable=0 violating=0
+----
+
+eval duration=10m
+----
+failed assertion sample 1
+  conformance unavailable=0 under=0 over=0 violating=0 
+  actual unavailable=0 under=0, over=0 violating=5
+violating constraints:
+  r1:000000{0000-2000} [(n1,s1):1, (n4,s4):4, (n3,s3):3, (n2,s2):5NON_VOTER] applying num_replicas=4 num_voters=3 constraints=[+region=a:2 +region=b:2] voter_constraints=[+region=a:2 +region=b:1]
+  r2:000000{2000-4000} [(n5,s5):1, (n1,s1):2, (n3,s3):4, (n2,s2):5NON_VOTER] applying num_replicas=4 num_voters=3 constraints=[+region=a:2 +region=b:2] voter_constraints=[+region=a:2 +region=b:1]
+  r3:000000{4000-6000} [(n5,s5):1, (n4,s4):2, (n2,s2):4, (n1,s1):5NON_VOTER] applying num_replicas=4 num_voters=3 constraints=[+region=a:2 +region=b:2] voter_constraints=[+region=a:2 +region=b:1]
+  r4:000000{6000-8000} [(n1,s1):4, (n4,s4):2, (n3,s3):3, (n2,s2):5NON_VOTER] applying num_replicas=4 num_voters=3 constraints=[+region=a:2 +region=b:2] voter_constraints=[+region=a:2 +region=b:1]
+  r5:00000{08000-10000} [(n2,s2):4, (n4,s4):2, (n3,s3):3, (n1,s1):5NON_VOTER] applying num_replicas=4 num_voters=3 constraints=[+region=a:2 +region=b:2] voter_constraints=[+region=a:2 +region=b:1]
+
+topology
+----
+a
+  └── [1 2]
+b
+  └── [3 4 5]
+
+# vim:ft=sh

--- a/pkg/kv/kvserver/asim/tests/testdata/non_rand/example_conformance
+++ b/pkg/kv/kvserver/asim/tests/testdata/non_rand/example_conformance
@@ -44,15 +44,7 @@ assertion type=conformance under=0 over=0 unavailable=0 violating=0
 
 eval duration=10m
 ----
-failed assertion sample 1
-  conformance unavailable=0 under=0 over=0 violating=0 
-  actual unavailable=0 under=0, over=0 violating=5
-violating constraints:
-  r1:000000{0000-2000} [(n1,s1):1, (n4,s4):4, (n3,s3):3, (n2,s2):5NON_VOTER] applying num_replicas=4 num_voters=3 constraints=[+region=a:2 +region=b:2] voter_constraints=[+region=a:2 +region=b:1]
-  r2:000000{2000-4000} [(n5,s5):1, (n1,s1):2, (n3,s3):4, (n2,s2):5NON_VOTER] applying num_replicas=4 num_voters=3 constraints=[+region=a:2 +region=b:2] voter_constraints=[+region=a:2 +region=b:1]
-  r3:000000{4000-6000} [(n5,s5):1, (n4,s4):2, (n2,s2):4, (n1,s1):5NON_VOTER] applying num_replicas=4 num_voters=3 constraints=[+region=a:2 +region=b:2] voter_constraints=[+region=a:2 +region=b:1]
-  r4:000000{6000-8000} [(n1,s1):4, (n4,s4):2, (n3,s3):3, (n2,s2):5NON_VOTER] applying num_replicas=4 num_voters=3 constraints=[+region=a:2 +region=b:2] voter_constraints=[+region=a:2 +region=b:1]
-  r5:00000{08000-10000} [(n2,s2):4, (n4,s4):2, (n3,s3):3, (n1,s1):5NON_VOTER] applying num_replicas=4 num_voters=3 constraints=[+region=a:2 +region=b:2] voter_constraints=[+region=a:2 +region=b:1]
+OK
 
 topology
 ----

--- a/pkg/kv/kvserver/replicate_queue_test.go
+++ b/pkg/kv/kvserver/replicate_queue_test.go
@@ -1646,6 +1646,7 @@ func TestReplicateQueueShouldQueueNonVoter(t *testing.T) {
 		" constraints='{\"+rack=3\": 1}', voter_constraints='{\"+rack=1\": 1}'")
 	require.NoError(t, err)
 
+	matchString := "rebalance target found for non-voter, enqueuing"
 	require.Eventually(t, func() bool {
 		// NB: Manually enqueuing the replica on server 0 (i.e. rack 1) is copacetic
 		// because we know that it is the leaseholder (since it is the only voting
@@ -1662,8 +1663,10 @@ func TestReplicateQueueShouldQueueNonVoter(t *testing.T) {
 			log.Errorf(ctx, "processErr: %s", processErr.Error())
 			return false
 		}
-		if matched, err := regexp.Match("rebalance target found for non-voter, enqueuing",
+		if matched, err := regexp.Match(matchString,
 			[]byte(recording.String())); !matched {
+			log.Infof(ctx, "didn't find matching string '%s' in trace %s",
+				matchString, recording.String())
 			require.NoError(t, err)
 			return false
 		}

--- a/pkg/kv/kvserver/replicate_queue_test.go
+++ b/pkg/kv/kvserver/replicate_queue_test.go
@@ -1578,6 +1578,10 @@ func TestReplicateQueueShouldQueueNonVoter(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 
+	// The zone config change leads to snapshot timeouts under stress race which
+	// make the test take 300+s.
+	skip.UnderStressRace(t)
+
 	ctx := context.Background()
 	serverArgs := make(map[int]base.TestServerArgs)
 	// Assign each store a rack number so we can constrain individual voting and

